<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>Hexo</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta property="og:type" content="website">
<meta property="og:title" content="Hexo">
<meta property="og:url" content="http://example.com/index.html">
<meta property="og:site_name" content="Hexo">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="John Doe">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="Hexo" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
  
<meta name="generator" content="Hexo 7.2.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Hexo</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"><span class="fa fa-bars"></span></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
        
          <a class="nav-icon" href="/atom.xml" title="RSS Feed"><span class="fa fa-rss"></span></a>
        
        <a class="nav-icon nav-search-btn" title="Search"><span class="fa fa-search"></span></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main">
  
    <article id="post-GATK4使用方法详解-变异检测" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2024/01/23/GATK4%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95%E8%AF%A6%E8%A7%A3-%E5%8F%98%E5%BC%82%E6%A3%80%E6%B5%8B/" class="article-date">
  <time class="dt-published" datetime="2024-01-23T06:08:14.000Z" itemprop="datePublished">2024-01-23</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8F/">二代测序</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2024/01/23/GATK4%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95%E8%AF%A6%E8%A7%A3-%E5%8F%98%E5%BC%82%E6%A3%80%E6%B5%8B/">GATK使用方法详解-变异检测</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>GATK4是最新的GATK版本，它在算法上进行了优化，运行速率得到提高，而且整合了picard。GATK4依然是用java 语言开发的，但使用方式上更加人性化，比如所有命令都是gatk cmd方式，这里的cmd是任何可以用的cmd。GATK4 的最佳实践给出了5套pipeline: <strong>Germline SNP&#x2F;Indel</strong>, Somatic SNV&#x2F;Indel, RNAseq SNP&#x2F;Indel, Germline CNV, Somatic CNV。目前版本号是 4.2.4 <a target="_blank" rel="noopener" href="https://gatk.broadinstitute.org/hc/en-us">https://gatk.broadinstitute.org/hc/en-us</a> </p>
<h1 id="1-建立索引"><a href="#1-建立索引" class="headerlink" title="1. 建立索引"></a>1. 建立索引</h1><ol>
<li>gatk  CreateSequenceDictionary -R hg19.fa -O hg19.dict</li>
<li>samtools faidx genome.fasta # .fai</li>
<li>bwa index ref.fa # .amb, .ann, .bwt, .pac, .sa</li>
<li>qplot –reference ucsc.hg19.fasta  –create_gc ucsc.hg19.winsize100.gc #.winsize100.gc</li>
<li>bowtie-build ucsc.hg19.fasta ucsc.hg19 ##</li>
<li>bowtie2-build human_g1k_v37.fa human_g1k_v37 #bowtie2-2.4.1</li>
</ol>
<h1 id="2-比对"><a href="#2-比对" class="headerlink" title="2. 比对"></a>2. 比对</h1><ul>
<li>BWA for DNA, STAR for RNA-seq</li>
</ul>
<h2 id="2-1-bwa-比对"><a href="#2-1-bwa-比对" class="headerlink" title="2.1 bwa 比对"></a>2.1 bwa 比对</h2><p>bwa mem -M -t 4 -R ‘@RG\tID:id1\tPL:illumina\tSM:test’ ref.fa test_1.fq test_2.fq | samtools view -bS - &gt;test.bam</p>
<p><strong>参数</strong><br>-R 设置reads group，gatk必须要的信息，其中ID，PL和SM信息是必须要的<br>GATK 要求read group的格式</p>
<h2 id="2-2-排序"><a href="#2-2-排序" class="headerlink" title="2.2 排序"></a>2.2 排序</h2><p>samtools sort -@ 3 -o test.sorted.bam test.bam<br>samtools index test.sorted.bam<br>rm -rf test.bam</p>
<p>ID &#x3D; Read group identifier<br>　　每一个read group 独有的ID，每一对reads 均有一个独特的ID，可以自定义命名；</p>
<p>PL &#x3D; Platform<br>　　测序平台；ILLUMINA, SOLID, LS454, HELICOS and PACBIO，不区分大小写；</p>
<p>SM &#x3D; sample<br>　　reads属于的样品名；SM要设定正确，因为GATK产生的VCF文件也使用这个名字;</p>
<p>LB &#x3D; DNA preparation library identifier<br>　　对一个read group的reads进行重复序列标记时，需要使用LB来区分reads来自那条lane;有时候，同一个库可能在不同的lane上完成测序;为了加以区分，</p>
<p>　　同一个或不同库只要是在不同的lane产生的reads都要单独给一个ID. 一般无特殊说明，成对儿read属于同一库，可自定义，比如：library1</p>
<p>若是忘记添加read group信息还以通过 AddOrReplaceReadGroups 添加</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">gatk AddOrReplaceReadGroups -I .bam -O .add.bam -LB library1 -PL illumina -PU pl1 -SM name</span><br><span class="line"></span><br><span class="line">##参数</span><br><span class="line">-I Input file (BAM or SAM or a GA4GH url)；</span><br><span class="line">-O  Output file (BAM or SAM)；</span><br><span class="line">-LB Read-Group library；</span><br><span class="line">-PL  Read-Group platform (e.g. ILLUMINA, SOLID)；</span><br><span class="line">-PU Read-Group platform unit (eg. run barcode)；</span><br><span class="line">-SM Read-Group sample name</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="2-3-标记重复序列"><a href="#2-3-标记重复序列" class="headerlink" title="2.3 标记重复序列"></a>2.3 标记重复序列</h2><p>gatk  MarkDuplicates -I test.sorted.bam -O test.sorted.markdup.bam -M test.sorted.markdup_metrics.txt –CREATE_INDEX true -VALIDATION_STRINGENCY LENIENT</p>
<p>rm -f test.sorted.bam test.sorted.bam.bai</p>
<p>##参数<br>-I 排序后的一个或者多个bam或者sam文件<br>-M 输出重复矩阵<br>-O 输出文件</p>
<h2 id="2-4-BQSR-BaseRecalibrator"><a href="#2-4-BQSR-BaseRecalibrator" class="headerlink" title="2.4 BQSR BaseRecalibrator"></a>2.4 BQSR BaseRecalibrator</h2><ul>
<li>所谓的变异位点，就是与参考基因组不同的部分，假设原始数据中就存在着一些由于测序仪器产生的系统性误差，那么变异位点识别过程中找到的variant，就会存在大量的假阳性。这一步主要目的是调整原始碱基的质量分数。</li>
</ul>
<p>gatk –java-options “-Xmx8g -XX:+UseParallelGC -XX:ParallelGCThreads&#x3D;1” BaseRecalibrator -R hg19.fa -I test.sorted.markdup.bam –known-sites $hash_config{“dbsnp”} –known-sites $hash_config{“Mills_and_1000G_gold_standard_indels”}  -O sample.recal.grp –default-base-qualities 1 2&gt;&gt;$log_dir/$sample.log 1&gt;&gt;$log_dir/$sample.log&#x2F;;  ##–known-sites $hash_config{“KG_phase1_indels”}</p>
<p>gatk –java-options “-Xmx8g -XX:+UseParallelGC -XX:ParallelGCThreads&#x3D;1” ApplyBQSR -R hg19.fa -I test.sorted.markdup.bam -bqsr sample.recal.grp -O sample.recal.bam</p>
<h2 id="2-5-检测变异"><a href="#2-5-检测变异" class="headerlink" title="2.5 检测变异"></a>2.5 检测变异</h2><p>##两种方法</p>
<p>##（1）多样本一起call，此次只有一个样本，若有多个样本，则继续用 -I 参数添加即可<br>gatk –java-options -Xmx4G HaplotypeCaller -I test.sorted.markup.bam -O test.gvcf1 -R ref.fa</p>
<h2 id="（2）单个样本call，然后在合并"><a href="#（2）单个样本call，然后在合并" class="headerlink" title="（2）单个样本call，然后在合并"></a>（2）单个样本call，然后在合并</h2><h2 id="生成中间文件gvcf"><a href="#生成中间文件gvcf" class="headerlink" title="生成中间文件gvcf"></a>生成中间文件gvcf</h2><p>gatk –java-options -Xmx4G HaplotypeCaller -I test.sorted.markup.bam -O test.gvcf -R ref.fa –emit-ref-confidence GVCF</p>
<p>gatk –java-options “-Xmx8g -XX:+UseParallelGC -XX:ParallelGCThreads&#x3D;1” HaplotypeCaller -R hg19.fa -I test.sorted.markup.recal.bam –dbsnp $hash_config{“dbsnp”} -O sample.1.g.vcf.gz –emit-ref-confidence GVCF</p>
<p>##通过gvcf检测变异, -V 添加上步得到的gvcf<br>gatk GenotypeGVCFs -R ref.fa -V test.gvcf -O test.vcf</p>
<p>##参数<br>-I BAM&#x2F;SAM&#x2F;CRAM file<br>-O  输出文件<br>-R 参考基因组<br>–java-options: 若设置java则需要添加<br>-Xmx4G：内存为4G，防止内存太大<br>-V  A VCF file containing variants</p>
<h1 id="2-6-qplot"><a href="#2-6-qplot" class="headerlink" title="2.6 qplot"></a>2.6 qplot</h1><h1 id="2-7-merge-gvcf"><a href="#2-7-merge-gvcf" class="headerlink" title="2.7 merge gvcf"></a>2.7 merge gvcf</h1><p>-按照每条染色体<br>gatk –java-options “-Xmx16g” CombineGVCFs -R hg19.fa $inputgvcf -O combine.1.g.vcf.gz -L 1 </p>
<h1 id="2-8-calling-variants"><a href="#2-8-calling-variants" class="headerlink" title="2.8 calling variants"></a>2.8 calling variants</h1><p>gatk –java-options “-Xmx8g -XX:+UseParallelGC -XX:ParallelGCThreads&#x3D;1” GenotypeGVCFs -R hg19.fa -V sample.1.g.vcf.gz -O sample.1.vcf.gz –dbsnp $hash_config{“dbsnp”} -L 1</p>
<h1 id="3-VQSR-or-hardFilter"><a href="#3-VQSR-or-hardFilter" class="headerlink" title="3. VQSR or hardFilter"></a>3. VQSR or hardFilter</h1><h2 id="3-1-VQSR"><a href="#3-1-VQSR" class="headerlink" title="3.1 VQSR"></a>3.1 VQSR</h2><p>#SNP calling</p>
<h1 id="rescource-文件需要先index，参考文件尽量全，最后三行的文件都是输出"><a href="#rescource-文件需要先index，参考文件尽量全，最后三行的文件都是输出" class="headerlink" title="rescource 文件需要先index，参考文件尽量全，最后三行的文件都是输出"></a>rescource 文件需要先index，参考文件尽量全，最后三行的文件都是输出</h1><h1 id="步骤1-4"><a href="#步骤1-4" class="headerlink" title="步骤1-4"></a>步骤1-4</h1><p>$ time gatk VariantRecalibrator -R &#x2F;path&#x2F;to&#x2F;hg38.fa -V 19P0126636WES.HC.vcf <br>-resource:hapmap,known&#x3D;false,training&#x3D;true,truth&#x3D;true,prior&#x3D;15.0 &#x2F;gatkdoc&#x2F;hapmap_3.3.hg38.vcf <br>-resource:omini,known&#x3D;false,training&#x3D;true,truth&#x3D;false,prior&#x3D;12.0 &#x2F;gatkdoc&#x2F;resources_broad_hg38_v0_1000G_omni2.5.hg38.vcf <br>-resource:1000G,known&#x3D;false,training&#x3D;true,truth&#x3D;false,prior&#x3D;10.0 &#x2F;gatkdoc&#x2F;resources_broad_hg38_v0_1000G_phase1.snps.high_confidence.hg38.vcf <br>-resource:dbsnp,known&#x3D;true,training&#x3D;false,truth&#x3D;false,prior&#x3D;2.0 &#x2F;gatkdoc&#x2F;resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf <br>-an DP -an QD -an FS -an SOR -an ReadPosRankSum -an MQRankSum -mode SNP -tranche 100.0 <br>-tranche 99.9 -tranche 99.0 -tranche 95.0 -tranche 90.0 <br>-O 19P0126636WES.snp.recal <br>–tranches-file 19P0126636WES.snp.tranches <br>–rscript-file 19P0126636WES.snp.plots.R</p>
<h1 id="步骤5"><a href="#步骤5" class="headerlink" title="步骤5"></a>步骤5</h1><p>$time gatk ApplyVQSR -R &#x2F;path&#x2F;to&#x2F;hg38.fa -V 19P0126636WES.HC.vcf <br>–ts-filter-level 99.0 –tranches-file 19P0126636WES.snp.tranches <br>–recal-file 19P0126636WES.snp.recal <br>-mode SNP<br>-O 19P0126636WES.snps.VQSR.vcf.gz</p>
<h2 id="查看数据文件"><a href="#查看数据文件" class="headerlink" title="查看数据文件"></a>查看数据文件</h2><p>$ cat 19P0126636WES.snp.tranches</p>
<h1 id="Variant-quality-score-tranches-file"><a href="#Variant-quality-score-tranches-file" class="headerlink" title="Variant quality score tranches file"></a>Variant quality score tranches file</h1><h1 id="Version-number-5"><a href="#Version-number-5" class="headerlink" title="Version number 5"></a>Version number 5</h1><p>targetTruthSensitivity,numKnown,numNovel,knownTiTv,novelTiTv,minVQSLod,filterName,model,accessibleTruthSites,callsAtTruthSites,truthSensitivity<br>90.00,1096434,69773,2.1008,1.6731,5.6910,VQSRTrancheSNP0.00to90.00,SNP,665207,598687,0.9000<br>95.00,1200002,98879,2.0923,1.6863,5.0278,VQSRTrancheSNP90.00to95.00,SNP,665207,631946,0.9500<br>99.00,1285374,122947,2.0888,1.6682,3.0695,VQSRTrancheSNP95.00to99.00,SNP,665207,658554,0.9900<br>99.90,1320241,151100,2.0868,1.6411,-1.4500,VQSRTrancheSNP99.00to99.90,SNP,665207,664541,0.9990<br>100.00,1334104,169658,2.0828,1.5982,-1179.7969,VQSRTrancheSNP99.90to100.00,SNP,665207,665207,1.0000</p>
<ul>
<li>-tranche默认是输出[100,99.9,99.0,90.0]4个tranche阈值的统计结果，如果想看其他阈值的结果，需要自行加上；结果就是看19P0126636WES.snp.tranches，而19P0126636WES.snp.recal文件则是用于ApplyVQSR的</li>
</ul>
<h3 id="INDEL-calling"><a href="#INDEL-calling" class="headerlink" title="INDEL calling"></a>INDEL calling</h3><p>#加了–max-gaussians 6用于设定Gaussians（clusters of variants that have similar properties）的数目，即减少聚类的组数，从而使得每个组的变异位点数目达到要求<br>$time gatk VariantRecalibrator -R &#x2F;path&#x2F;to&#x2F;hg38.fa -V 19P0126636WES.snps.VQSR.vcf.gz <br>-resource:mills,known&#x3D;true,training&#x3D;true,truth&#x3D;true,prior&#x3D;12.0 &#x2F;gatkdoc&#x2F;resources_broad_hg38_v0_Mills_and_1000G_gold_standard.indels.hg38.vcf <br>-an DP -an QD -an FS -an SOR -an ReadPosRankSum -an MQRankSum <br>-mode INDEL –max-gaussians 6 <br>–rscript-file 19P0126636WES_L1.snp.indel.plots.R <br>–tranches-file 19P0126636WES.snp.indel.tranches <br>-O data&#x2F;19P0126636WES.snp.indel.recal</p>
<p>$ time gatk ApplyVQSR -R &#x2F;path&#x2F;to&#x2F;hg38.fa -V data&#x2F;19P0126636WES.snps.VQSR.vcf.gz <br>–ts-filter-level 99.0 <br>–tranches-file 19P0126636WES.snp.indel.tranches –recal-file 19P0126636WES.snp.indel.recal <br>-mode INDEL <br>-O 19P0126636WES.VQSR.vcf.gz<br>（上面这个就是我们千辛万苦想得到的文件啦！）</p>
<p>VQSR大概步骤：</p>
<ol>
<li>GATK认为VQSR比根据各种annotations进行hard-filtering过滤要好，减少了人为阈值的局限性，避免了一刀切的弊端，从而在sensitivity和specificity之间达到一定的平衡</li>
<li>VQSR根据机器学习算法从highly validated变异位点数据集（每个位点的annotation profile，一般使用5-8个annotation）中获取到good variants&#x2F;bad variants</li>
<li>根据上述的位点从我们自己数据集中挑选出一个变异子集（probably true positives）来建模训练，获得一个可识别good variants的模型；bad variants的模型也是如此获得</li>
<li>然后根据上述获得的模型，对自己数据集的每个变异位点进行一个总的打分</li>
<li>最后根据设定的sensitivity阈值对变异位点进行过滤</li>
</ol>
<p>按照官方教程，<br>SNP的VQSR过滤，选用的resource datasets为：</p>
<ul>
<li>HapMap，hapmap_3.3.hg38.vcf.gz，truth&#x3D;true表示VQSR将这个数据集中的变异位点作为真位点true sites，training&#x3D;true表示VQSR将true sites用于训练recalibration model，并赋予这些变异位点prior likelihood值为Q15 (96.84%)</li>
<li>Omni，1000G_omni2.5.hg38.vcf.gz，truth&#x3D;true，training&#x3D;false（文档中写着是true，参数建议中写着的是false。。。我就按照参数上的来了），Q12 (93.69%)</li>
<li>1000G，1000G_phase1.snps.high_confidence.hg38.vcf.gz，truth&#x3D;false表示VQSR考虑到在1000G数据集中的不仅包含了true variants还有false positives，training&#x3D;true，Q10 (90%)</li>
<li>dbSNP，dbsnp_146.hg38.vcf.gz&#x2F;dbsnp_138.hg38.vcf.gz，truth&#x3D;false表示VQSR未将dbSNP数据集中的位点作为可信数据集，training&#x3D;false表示不用于训练数据集，known&#x3D;true表示stratify output metrics such as Ti&#x2F;Tv ratio by whether variants are present in dbsnp or not，Q2 (36.90%)</li>
</ul>
<p>INDEL的VQSR过滤，选用的resource datasets为：</p>
<ul>
<li>Mills，Mills_and_1000G_gold_standard.indels.hg38.vcf.gz，truth&#x3D;true，training&#x3D;true，Q12 (93.69%)</li>
<li>dbSNP，dbsnp_146.hg38.vcf.gz&#x2F;dbsnp_138.hg38.vcf.gz，truth&#x3D;false，training&#x3D;false，known&#x3D;true，Q2 (36.90%)</li>
</ul>
<h2 id="3-2-gatk-VariantFiltration-O-test-snp-fil-vcf-temp-V-test-snp-vcf-–filter-expression-‘QUAL-30-0-QD-2-0-FS-60-0-SOR-4-0’-–filter-name-lowQualFilter-–cluster-window-size-10-–cluster-size-3-–missing-values-evaluate-as-failing"><a href="#3-2-gatk-VariantFiltration-O-test-snp-fil-vcf-temp-V-test-snp-vcf-–filter-expression-‘QUAL-30-0-QD-2-0-FS-60-0-SOR-4-0’-–filter-name-lowQualFilter-–cluster-window-size-10-–cluster-size-3-–missing-values-evaluate-as-failing" class="headerlink" title="3.2 gatk VariantFiltration -O test.snp.fil.vcf.temp -V test.snp.vcf –filter-expression ‘QUAL &lt; 30.0 || QD &lt; 2.0 || FS &gt; 60.0 ||  SOR &gt; 4.0’ –filter-name lowQualFilter –cluster-window-size 10  –cluster-size 3 –missing-values-evaluate-as-failing"></a>3.2 gatk VariantFiltration -O test.snp.fil.vcf.temp -V test.snp.vcf –filter-expression ‘QUAL &lt; 30.0 || QD &lt; 2.0 || FS &gt; 60.0 ||  SOR &gt; 4.0’ –filter-name lowQualFilter –cluster-window-size 10  –cluster-size 3 –missing-values-evaluate-as-failing</h2><h2 id="参数"><a href="#参数" class="headerlink" title="参数"></a>参数</h2><p>-O 输出filt.vcf文件<br>-V 输入vcf文件<br>–filter-expression 过滤条件, VCF INFO 信息<br>–cluster-window-size 以10个碱基为一个窗口<br>–cluster-size 10个碱基为窗口，若存在3以上个则过滤<br>–filter-name 被过滤掉的SNP不会删除，而是给一个标签， 比如 Filter<br>–missing-values-evaluate-as-failing 当筛选标准比较多的时候，可能有一些位点没有筛选条件当中的一条或几条，例如下面的这个表达式；QUAL &lt; 30.0 || QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || HaplotypeScore &gt; 13.0 并不一定所有位点都有这些信息，这种情况下GATK运行的时候会报很多WARNING信息，用这个参数可以把这些缺少某些FLAG的位点也给标记成没有通过筛选的。</p>
<h1 id="4-annovar"><a href="#4-annovar" class="headerlink" title="4. annovar"></a>4. annovar</h1><p>################<br>##4.2 ANNOVOR<br>################<br>###annovar database download<br>#gene-based<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar refGene humandb ## RefSeq does not have mitochondria gene definition<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar knownGene humandb ## UCSC Known Gene<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar ensGene humandb # Ensembl Gene<br>.&#x2F;annotate_variation.pl -build hg19 -downdb wgEncodeGencodeBasicV19  humandb #pass</p>
<p>#region-based<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb phastConsElements46way humandb&#x2F; # Conserved by 46-way<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb phastConsElements100way humandb&#x2F; # by 100-way alignment<br>.&#x2F;annotate_variation.pl -build hg19 -downdb tfbsConsSites humandb&#x2F; # Transcription factor binding site annotation<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb cytoBand humandb&#x2F; # represent cytogenetic bands<br>.&#x2F;annotate_variation.pl -build hg19 -downdb wgRna humandb&#x2F; # Identify variants disrupting microRNAs and snoRNAs<br>.&#x2F;annotate_variation.pl -build hg19 -downdb targetScanS humandb&#x2F; # Identify variants disrupting predicted microRNA binding sites<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb genomicSuperDups humandb&#x2F; # Identify variants located in segmental duplications<br>.&#x2F;annotate_variation.pl -build hg19 -downdb dgvMerged humandb&#x2F; # Identify previously reported structural variants in DGV (Database of Genomic Variants). deletions and duplications<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb gwasCatalog humandb&#x2F; # Identify variants reported in previously published GWAS</p>
<p>#now only for hg18. Identify variants in ENCODE annotated regions (transcribed regions, H3K4Me1 regions, H3K4Me3 regions, H3K27Ac regions, DNaseI hypersensitivity regions, transcription factor ChIP-Seq regions, etc)<br>.&#x2F;annotate_variation.pl -downdb wgEncodeCaltechRnaSeqRawSignalRep1Gm12878CellLongpolyaBb12x75 humandb&#x2F;<br>.&#x2F;annotate_variation.pl -downdb wgEncodeBroadChipSeqPeaksGm12878H3k4me1 humandb&#x2F;<br>.&#x2F;annotate_variation.pl -downdb wgEncodeRegDnaseClustered humandb&#x2F;<br>.&#x2F;annotate_variation.pl -downdb wgEncodeRegTfbsClustered humandb&#x2F;<br>.&#x2F;annotate_variation.pl -downdb wgEncodeBroadHmmGm12878HMM humandb&#x2F; #Identify non-coding variants that disrupt enhancers, repressors, promoters<br>.&#x2F;annotate_variation.pl -downdb wgEncodeUwDnaseSeqHotspotsRep2Gm12878 humandb&#x2F; #Identify variants in other genomic regions annotated with other functions</p>
<p>#filter-based<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb 1000g2015aug humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar avsnp150  humandb&#x2F;<br>.&#x2F;annotate_variation.pl  -build hg19 -downdb -webfrom annovar popfreq_all humandb&#x2F; #?? PopFreqMax 1000G2012APR_ALL 1000G2012APR_AFR 1000G2012APR_AMR 1000G2012APR_ASN 1000G2012APR_EUR ESP6500si_ALL ESP6500si_AA ESP6500si_EA CG46.<br>.&#x2F;annotate_variation.pl  -build hg19 -downdb -webfrom annovar nci60 humandb&#x2F;<br>.&#x2F;annotate_variation.pl  -build hg19 -downdb -webfrom annovar popfreq_max humandb&#x2F; </p>
<p>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar dbnsfp35a humandb<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar dbnsfp31a_interpro humandb<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar dbscsnv11 humandb<br>.&#x2F;annotate_variation.pl -downdb -buildver hg19 -webfrom annovar gerp++gt2 humandb&#x2F;</p>
<p>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar esp6500siv2_all humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar esp6500siv2_aa humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar esp6500siv2_ea humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar clinvar_20200316 humandb<br>或者 <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/">ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/</a> # <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37//weekly/clinvar_20220115.vcf.gz">ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37//weekly/clinvar_20220115.vcf.gz</a></p>
<p>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar cadd13gt10 humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar cosmic70 humandb&#x2F;</p>
<p>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar exac03 humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar exac03nontcga humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar exac03nonpsych humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar kaviar_20150923 humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar hrcr1 humandb&#x2F;<br>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar intervar_20180118 humandb&#x2F;</p>
<p>.&#x2F;annotate_variation.pl -buildver hg19 -downdb -webfrom annovar mitimpact24 humandb&#x2F;</p>
<p>perl ..&#x2F;prepare_annovar_user.pl -dbtype cosmic CosmicMutantExport.tsv  -vcf CosmicCodingMuts.vcf &gt; hg38_cosmic91.txt<br>perl ..&#x2F;prepare_annovar_user.pl -dbtype cosmic CosmicNCV.tsv  -vcf CosmicNonCodingVariants.vcf &gt; hg38_cosmic91.txt.noncoding</p>
<p>cosmic,clinvar</p>
<p>&#x2F;scratch&#x2F;cgg&#x2F;software&#x2F;annovar&#x2F;2014Jul14&#x2F;table_annovar.pl vcf.avi &#x2F;scratch&#x2F;cgg&#x2F;software&#x2F;annovar&#x2F;2014Jul14&#x2F;humandb&#x2F; -buildver hg19 -out myanno2 -remove -protocol refGene,cytoBand,genomicSuperDups,dgvMerged,popfreq_all,snp138,ljb23_all,caddgt10,cosmic68,clinvar_20140702,gwasCatalog,phastConsElements100way,tfbsConsSites,wgRna,targetScanS  -operation g,r,r,r,f,f,f,f,f,f,r,r,r,r,r -nastring . -csvout</p>
<p>#cg69<br>#cg46<br>nci60<br>popfreq_all<br>gerp++elem<br>gerp++gt2<br>####<br>1.gene-based<br>    refGene;knownGene;ensGene;<br>2.region-based<br>    phastConsElements46way;phastConsElements100way;tfbsConsSites;cytoBand;wgRna;genomicSuperDups;	</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2024/01/23/GATK4%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95%E8%AF%A6%E8%A7%A3-%E5%8F%98%E5%BC%82%E6%A3%80%E6%B5%8B/" data-id="clw7l0lmt0004olgp4z4dfjh7" data-title="GATK使用方法详解-变异检测" class="article-share-link"><span class="fa fa-share">Share</span></a>
      
      
      
    </footer>
  </div>
  
</article>



  
    <article id="post-VCF文件格式" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2024/01/22/VCF%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/" class="article-date">
  <time class="dt-published" datetime="2024-01-22T09:08:14.000Z" itemprop="datePublished">2024-01-22</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2024/01/22/VCF%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/">VCF文件格式详解</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p><strong>VCF文件格式详解</strong></p>
<p>VCF介绍</p>
<p>VCF是用于描述SNP，INDEL和SV结果的文本文件。在GATK软件中得到最好的支持，当然samtools得到的结果也是VCF格式。</p>
<h1 id="1-VCF的主体结构"><a href="#1-VCF的主体结构" class="headerlink" title="1. VCF的主体结构"></a>1. VCF的主体结构</h1><p>VCF文件分为两部分内容:</p>
<ul>
<li><ol>
<li>以“#”开头的注释部分</li>
</ol>
</li>
<li><ol start="2">
<li>没有“#”开头的主体部分</li>
</ol>
</li>
</ul>
<p>去掉了头部的注释行，只留下了代表每一行意义的注释行。<br>主体部分中每一行代表一个Variant的信息。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##fileformat=VCFv4.2</span></span><br><span class="line"><span class="comment">##fileDate=20090805</span></span><br><span class="line"><span class="comment">##source=myImputationProgramV3.1</span></span><br><span class="line"><span class="comment">##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta</span></span><br><span class="line"><span class="comment">##contig=&lt;ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d618ff66beb2da,species=&quot;Homo sapiens&quot;,taxonomy=x&gt;</span></span><br><span class="line"><span class="comment">##phasing=partial</span></span><br><span class="line"><span class="comment">##INFO=&lt;ID=NS,Number=1,Type=Integer,Description=&quot;Number of Samples With Data&quot;&gt;</span></span><br><span class="line"><span class="comment">##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Total Depth&quot;&gt;</span></span><br><span class="line"><span class="comment">##INFO=&lt;ID=AF,Number=A,Type=Float,Description=&quot;Allele Frequency&quot;&gt;</span></span><br><span class="line"><span class="comment">##INFO=&lt;ID=AA,Number=1,Type=String,Description=&quot;Ancestral Allele&quot;&gt;</span></span><br><span class="line"><span class="comment">##INFO=&lt;ID=DB,Number=0,Type=Flag,Description=&quot;dbSNP membership, build 129&quot;&gt;</span></span><br><span class="line"><span class="comment">##INFO=&lt;ID=H2,Number=0,Type=Flag,Description=&quot;HapMap2 membership&quot;&gt;</span></span><br><span class="line"><span class="comment">##FILTER=&lt;ID=q10,Description=&quot;Quality below 10&quot;&gt;</span></span><br><span class="line"><span class="comment">##FILTER=&lt;ID=s50,Description=&quot;Less than 50% of samples have data&quot;&gt;</span></span><br><span class="line"><span class="comment">##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;</span></span><br><span class="line"><span class="comment">##FORMAT=&lt;ID=GQ,Number=1,Type=Integer,Description=&quot;Genotype Quality&quot;&gt;</span></span><br><span class="line"><span class="comment">##FORMAT=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Read Depth&quot;&gt;</span></span><br><span class="line"><span class="comment">##FORMAT=&lt;ID=HQ,Number=2,Type=Integer,Description=&quot;Haplotype Quality&quot;&gt;</span></span><br><span class="line"><span class="comment">#CHROM POS ID REF ALT QUAL FILTER INFO  FORMAT  NA00001 NA00002</span></span><br><span class="line"><span class="number">1</span> <span class="number">899282</span> rs28548431 C T <span class="number">71.77</span> PASS AC=<span class="number">1</span>;AF=<span class="number">0</span>.<span class="number">50</span>;AN=<span class="number">2</span>;DB;DP=<span class="number">4</span>;Dels=<span class="number">0</span>.<span class="number">00</span>;HRun=<span class="number">0</span>;HaplotypeScore=<span class="number">0</span>.<span class="number">00</span>;MQ=<span class="number">99.00</span>;MQ<span class="number">0</span>=<span class="number">0</span>;QD=<span class="number">17.94</span>;SB=-<span class="number">46.55</span>;VQSLOD=-<span class="number">1.9148</span> GT:AD:DP:GQ:PL <span class="number">0</span>/<span class="number">1</span>:<span class="number">1</span>,<span class="number">3</span>:<span class="number">4</span>:<span class="number">25.92</span>:<span class="number">103</span>,<span class="number">0</span>,<span class="number">26</span></span><br><span class="line"><span class="number">20</span> <span class="number">14370</span> rs6054257 G A <span class="number">29</span> PASS NS=<span class="number">3</span>;DP=<span class="number">14</span>;AF=<span class="number">0</span>.<span class="number">5</span>;DB;H2 GT:GQ:DP:HQ <span class="number">0</span>|<span class="number">0</span>:<span class="number">48</span>:<span class="number">1</span>:<span class="number">51</span>,<span class="number">51</span> <span class="number">1</span>|<span class="number">0</span>:<span class="number">48</span>:<span class="number">8</span>:<span class="number">51</span>,<span class="number">51</span></span><br><span class="line"><span class="number">20</span> <span class="number">17330</span>  .        T A  <span class="number">3</span> q1<span class="number">0</span>  NS=<span class="number">3</span>;DP=<span class="number">11</span>;AF=<span class="number">0</span>.<span class="number">017</span> GT:GQ:DP:HQ <span class="number">0</span>|<span class="number">0</span>:<span class="number">49</span>:<span class="number">3</span>:<span class="number">58</span>,<span class="number">50</span> <span class="number">0</span>|<span class="number">1</span>:<span class="number">3</span>:<span class="number">5</span>:<span class="number">65</span>,<span class="number">3</span></span><br><span class="line"><span class="number">20</span> <span class="number">1110696</span> rs6040355 A G,T <span class="number">67</span> PASS NS=<span class="number">2</span>;DP=<span class="number">10</span>;AF=<span class="number">0</span>.<span class="number">333</span>,<span class="number">0</span>.<span class="number">667</span>;AA=T;DB GT:GQ:DP:HQ <span class="number">1</span>|<span class="number">2</span>:<span class="number">21</span>:<span class="number">6</span>:<span class="number">23</span>,<span class="number">27</span> <span class="number">2</span>|<span class="number">1</span>:<span class="number">2</span>:<span class="number">0</span>:<span class="number">18</span>,<span class="number">2</span></span><br><span class="line"><span class="number">20</span> <span class="number">1230237</span> . T  .  <span class="number">47</span> PASS NS=<span class="number">3</span>;DP=<span class="number">13</span>;AA=T GT:GQ:DP:HQ <span class="number">0</span>|<span class="number">0</span>:<span class="number">54</span>:<span class="number">7</span>:<span class="number">56</span>,<span class="number">60</span> <span class="number">0</span>|<span class="number">0</span>:<span class="number">48</span>:<span class="number">4</span>:<span class="number">51</span>,<span class="number">51</span></span><br></pre></td></tr></table></figure>

<ul>
<li>header: 1.基因组的版本(<strong>hg19,hg38</strong>) 2. info,filter,format注释描述</li>
<li>CHROM：表示变异位点是在哪个contig 里call出来的，如果是人类全基因组的话那就是chr1…chr22，chrX,Y,M。</li>
<li>POS：变异位点相对于参考基因组所在的位置，如果是indel，就是第一个碱基所在的位置。</li>
<li>ID：如果call出来的SNP存在于dbSNP数据库里，就会显示相应的dbSNP里的rs编号。</li>
<li>REF和REF：在这个变异位点处，参考基因组中所对应的碱基和研究对象基因组中所对应的碱基。</li>
<li>QUAL：可以理解为所call出来的变异位点的质量值。Q&#x3D;-10lgP，Q表示质量值；P表示这个位点发生错误的概率。因此，如果想把错误率从控制在90%以上，P的阈值就是1&#x2F;10，那lg（1&#x2F;10）&#x3D;-1，Q&#x3D;（-10）*（-1）&#x3D;10。同理，当Q&#x3D;20时，错误率就控制在了0.01。</li>
<li>FILTER：理想情况下，QUAL这个值应该是用所有的错误模型算出来的，这个值就可以代表正确的变异位点了，但是事实是做不到的。因此，还需要对原始变异位点做进一步的过滤。无论你用什么方法对变异位点进行过滤，过滤完了之后，在FILTER一栏都会留下过滤记录，如果是通过了过滤标准，那么这些通过标准的好的变异位点的FILTER一栏就会注释一个PASS，如果没有通过过滤，就会在FILTER这一栏提示除了PASS的其他信息。如果这一栏是一个“.”的话，就说明没有进行过任何过滤。GATK的<strong>VQSR</strong>可以注释。</li>
<li><strong>INFO</strong> 这一行是variant的详细信息，内容很多，以下再具体详述。一般可以含有annovar注释，GATK注释等信息。</li>
<li><strong>FORMAT</strong>: 在<strong>header</strong>里面都有其注解。<ul>
<li><p><strong>GT</strong>：表示这个样本的基因型，对于一个二倍体生物，GT值表示的是这个样本在这个位点所携带的两个等位基因。0表示跟REF一样；1表示表示跟ALT一样；2表示第二个ALT。当只有一个ALT等位基因的时候，0&#x2F;0表示纯和且跟REF一致；0&#x2F;1表示杂合，两个allele一个是ALT一个是REF；1&#x2F;1表示纯和且都为ALT；</p>
<ul>
<li>0 - reference call</li>
<li>1 - alternative call 1</li>
<li>2 - alternative call 2</li>
<li>The allele separator is ‘&#x2F;‘ for unphased genotypes and ‘|’ for phased genotypes.</li>
</ul>
</li>
<li><p><strong>AD</strong>：对应两个以逗号隔开的值，这两个值分别表示覆盖到REF和ALT碱基的reads数，相当于支持REF和支持ALT的测序深度。</p>
</li>
<li><p><strong>DP</strong>：覆盖到这个位点的总的reads数量，相当于这个位点的深度。</p>
</li>
<li><p><strong>PL</strong>: 对应3个以逗号隔开的值，这三个值分别表示该位点基因型是0&#x2F;0，0&#x2F;1，1&#x2F;1的没经过先验的标准化Phred-scaled似然值（L）。如果转换成支持该基因型概率（P）的话，由于L&#x3D;-10lgP，那么P&#x3D;10^（-L&#x2F;10），因此，当L值为0时，P&#x3D;10^0&#x3D;1。因此，这个值越小，支持概率就越大，也就是说是这个基因型的可能性越大。</p>
</li>
<li><p><strong>GQ</strong>：表示最可能的基因型的质量值。表示的意义同QUAL。</p>
</li>
<li><p>其他：</p>
<ul>
<li>FS: FisherStrand的缩写，表示使用Fisher’s精确检验来检测strand bias而得到的Fhred格式的p值，该值越小越好；如果该值较大，表示strand bias（正负链偏移）越严重，即所检测到的variants位点上，reads比对到正负义链上的比例不均衡。一般进行filter的时候，推荐保留FS&lt;10-20的variants位点。GATK可设定FS参数。</li>
<li>ReadPosRandSum: Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias.当variants出现在reads尾部的时候，其结果可能不准确。该值用于衡量alternative allele（变异的等位基因）相比于reference allele（参考基因组等位基因），其variant位点是否匹配到reads更靠中部的位置。因此只有基因型是杂合且有一个allele和参考基因组一致的时候，才能计算该值。若该值为正值，表明和alternative allele相当于reference allele，落来reads更靠中部的位置；若该值是负值，则表示alternative allele相比于reference allele落在reads更靠尾部的位置。 进行filter的之后，推荐保留ReadPosRankSum&gt;-1.65 - -3.0的variant位点</li>
<li>MQRankSum: 该值用于衡量alternative allele上reads的mapping quality与reference allele上reads的mapping quality的差异。若该值是负数值，则表明alternative allele比reference allele的reads mapping quality差。进行filter的时候，推荐保留MQRankSum&gt;-1.65 - -3.0的variant位点。</li>
<li>HaplotypeScore： Consistency of the site with at most two segregating haplotypes</li>
<li>InbreedingCoeff： Inbreeding coefficient as estimated from the genotype likelihoods per-sample when compared against the Hard-Weinberg expectation</li>
<li>MLEAC：Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed</li>
<li>MLEAF：Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT alle in the same order as listed</li>
<li>MQ：RMS Mapping Quality</li>
<li>MQ0：Total Mapping Quality Zero Reads</li>
<li>MQRankSum： Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities</li>
<li>QD：Variant Confidence&#x2F;Quality by Depth</li>
<li>RPA：Number of times tandem repeat unit is repeated, for each allele (including reference)</li>
<li>RU：Tandem repeat unit (bases)</li>
<li>ReadPosRankSum：Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias</li>
<li>STR： Variant is a short tandem repeat</li>
</ul>
</li>
</ul>
</li>
</ul>
<h1 id="2-重要软件-tabix-bgzip"><a href="#2-重要软件-tabix-bgzip" class="headerlink" title="2. 重要软件 tabix bgzip"></a>2. 重要软件 tabix bgzip</h1><ul>
<li><strong>tabix</strong> 可以对NGS分析中常见格式的文件建立索引，从而加快访问速度，不仅支持VCF文件，还支持BED, GFF，SAM等格式。</li>
</ul>
<p>由于SNP位点数量巨大，对应VCF文件也非常的大，为例节省存储空间，最常见的做法就是压缩。<strong>bgzip</strong>可以压缩VCF文件，用法如下</p>
<blockquote>
<p>bgzip  view.vcf</p>
</blockquote>
<p>压缩之后，原本的<strong>view.vcf</strong>文件就变成了<strong>view.vcf.gz</strong>文件。压缩后缀为.gz, 如果想要解压缩，有以下两种用法</p>
  <figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">bgzip -d view.vcf.gz</span><br><span class="line">gunzip view.vcf.gz</span><br><span class="line">gunzip -c view.vcf.gz | less</span><br><span class="line">tabix view.vcf.gz 1 |less</span><br></pre></td></tr></table></figure>
<p><strong>bgzip</strong>的压缩算法和<strong>gzip</strong>压缩算法有着相似之处，所以对于bgzip压缩的文件，解压缩时除了可以使用bgzip软件本身，还可以使用gunzip进行解压缩。</p>
<p>需要注意的是，两种算法虽然有相似之处，但是还是有本质区别的，在对VCF文件压缩时，不可以使用gzip来代替bgzip。</p>
<p>对于大型的VCF文件而言，如何快速访问其中的记录也是个难点。<strong>tabix</strong>可以对VCF文件构建索引，索引构建好之后，访问速度会快很多。<strong>tabix</strong>对VCF文件建立索引的用法如下</p>
<blockquote>
<p>tabix -p vcf view.vcf.gz</p>
</blockquote>
<blockquote>
<p>tabix -s 1 -b 2 -e 2 vcf view.vcf.gz</p>
</blockquote>
<p>注意输入的VCF文件必须是使用bgzip压缩之后的VCF文件，生成的索引文件为view.vcf.gz.tbi, 后缀为.tbi。</p>
<p>构建好索引之后，可以快速的获取指定区域的记录，示例如下</p>
<ol>
<li><p>获取位于11号染色体的SNP位点</p>
<blockquote>
<p>tabix view.vcf.gz 11</p>
</blockquote>
</li>
<li><p>获取位于11号染色体上突变位置大于或者等于2343545的SNP位点</p>
<blockquote>
<p>tabix view.vcf.gz 11:2343545</p>
</blockquote>
</li>
<li><p>获取位于11号染色体上突变位置介于2343540到2343596的SNP位点</p>
<blockquote>
<p>tabix view.vcf.gz 11:2343540-2343596</p>
</blockquote>
</li>
<li><p>获取位多个SNP位点</p>
<blockquote>
<p>tabix view.vcf.gz 11:2343540-2343596 12:2343540-2343596</p>
</blockquote>
</li>
</ol>
<p>很多操作VCF的软件都会识别tabix建立的索引，从而加快处理速度。很多大型项目VCF文件，也都会用bgzip压缩，然后建立tabix的索引。</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2024/01/22/VCF%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/" data-id="clw7l0lmt0005olgp63ov2l0v" data-title="VCF文件格式详解" class="article-share-link"><span class="fa fa-share">Share</span></a>
      
      
      
    </footer>
  </div>
  
</article>



  
    <article id="post-bam文件格式" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2024/01/22/bam%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/" class="article-date">
  <time class="dt-published" datetime="2024-01-22T08:08:14.000Z" itemprop="datePublished">2024-01-22</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2024/01/22/bam%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/">Sam/Bam文件格式详解</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p><strong>Sam&#x2F;Bam文件格式详解</strong></p>
<p>bam文件是sam文件的二进制格式。(<a target="_blank" rel="noopener" href="https://samtools.github.io/hts-specs/SAMv1.pdf">https://samtools.github.io/hts-specs/SAMv1.pdf</a>)</p>
<h1 id="1-Header-section"><a href="#1-Header-section" class="headerlink" title="1. Header section"></a>1. Header section</h1><ul>
<li>该部分全部以“@”开头，提供基本的软件版本，参考序列信息，排序信息等</li>
<li>@HD行：这一行中有各种不同的标识</li>
<li>标识“VN”用以说明格式版本</li>
<li>标识“SO”用以说明比对排序的情况，有unknown (default)、unsorted、queryname和coordinate,对于coordinate，排序的主键是Alignments section的第三列“RNAME”，其顺序由@SQ行的“SN”标识的顺序定义，次要排序键是Alignments section的第四列“POS”字段。对于RNAME和POS相等的比对，排列顺序则是任意的</li>
<li>@SQ行的“SN”标签是参考序列说明，它的值主要是用于Alignments section的第三列“RNAME”和第七列“MRNM”比对的记录</li>
<li>@PG行是使用的程序说明；该行“ID”为程序记录标识符，“PN”为程序名字，“CL”为命令行</li>
<li>@CO行是任意的说明信息</li>
</ul>
<p><strong>注意事项</strong></p>
<ol>
<li><strong>mapping(bwa,GATK,bismark,…)</strong></li>
<li><strong>duplication(picard,samblaster,sambamba,…)</strong></li>
<li><strong>reference genome(hg19,hg38)</strong></li>
</ol>
<h1 id="2-Alignments-section"><a href="#2-Alignments-section" class="headerlink" title="2. Alignments section"></a>2. Alignments section</h1><p>该部分包含了11列必需字段，无效或者没有的字段一般用“0”或者“*”表示</p>
<table>
<thead>
<tr>
<th align="left">Col</th>
<th align="left">Field</th>
<th align="center">Type</th>
<th align="center">Regexp&#x2F;Range</th>
<th align="left">Brief description</th>
</tr>
</thead>
<tbody><tr>
<td align="left">1</td>
<td align="left">QNAME</td>
<td align="center">String</td>
<td align="center">[!-?A-~]{1,254}</td>
<td align="left">Query template NAME</td>
</tr>
<tr>
<td align="left">2</td>
<td align="left">FLAG</td>
<td align="center">Int</td>
<td align="center">[0,$2^{16}$−1]</td>
<td align="left">bitwise FLAG</td>
</tr>
<tr>
<td align="left">3</td>
<td align="left">RNAME</td>
<td align="center">String</td>
<td align="center">&#96;&#96;&#96;*</td>
<td align="left">[:rname: <em>&#x3D;][:rname:]</em>&#96;&#96;&#96;</td>
</tr>
<tr>
<td align="left">4</td>
<td align="left">POS</td>
<td align="center">Int</td>
<td align="center">[0,$2^{31}$−1]</td>
<td align="left">POSition</td>
</tr>
<tr>
<td align="left">5</td>
<td align="left">MAPQ</td>
<td align="center">Int</td>
<td align="center">[0,$2^8$−1]</td>
<td align="left">MAPping Quality</td>
</tr>
<tr>
<td align="left">6</td>
<td align="left">CIGAR</td>
<td align="center">String</td>
<td align="center">\*|([0-9]+[MIDNSHPX&#x3D;])+</td>
<td align="left">CIGAR string</td>
</tr>
<tr>
<td align="left">7</td>
<td align="left">RNEXT</td>
<td align="center">String</td>
<td align="center">\*|&#x3D;|[:rname:*&#x3D;][:rname:]*</td>
<td align="left">Reference name of the mate&#x2F;next read</td>
</tr>
<tr>
<td align="left">8</td>
<td align="left">PNEXT</td>
<td align="center">Int</td>
<td align="center">[0,$2^{31}$−1]</td>
<td align="left">Position of the mate&#x2F;next read</td>
</tr>
<tr>
<td align="left">9</td>
<td align="left">TLEN</td>
<td align="center">Int</td>
<td align="center">[−$2^{31}$+1,$2^{31}$−1]</td>
<td align="left">observed Template LENgth</td>
</tr>
<tr>
<td align="left">10</td>
<td align="left">SEQ</td>
<td align="center">String</td>
<td align="center">\*|[A-Za-z&#x3D;.]+</td>
<td align="left">segment SEQuence</td>
</tr>
<tr>
<td align="left">11</td>
<td align="left">QUAL</td>
<td align="center">String</td>
<td align="center">[!-~]+ (U+2227)</td>
<td align="left">ASCII of Phred-scaled base QUALity+33</td>
</tr>
</tbody></table>
<ul>
<li>1.QNAME 序列的名字，也就是reads的名称 </li>
<li>2.<strong>FLAG</strong> 是一个标记的数字，是有需要转换成二进制才能知道代表的意思，各个数字分别代表</li>
<li>3.RNAME 参考序列的名字 </li>
<li>4.POS 在参考序列上的位置 </li>
<li>5.MAPQ mapping qulity 越高则位点越独特，比对的质量值</li>
<li>6.<strong>CIGAR</strong> 代表比对结果的CIGAR字符串，如37M1D2M1I，这段字符的意思是37个匹配，1个参考序列上的删除，2个匹配，1个参考序列上的插入。M代表的是alignment match(可以是错配)，可以理解为表示比对的具体情况 </li>
<li>7.RNEXT mate 序列所在参考序列的名称，mate一般指大的片段序列 </li>
<li>8.PNEXT mate 序列在参考序列上的位置 </li>
<li>9.<strong>TLEN</strong> 估计出的片段的长度，当mate 序列位于本序列上游时该值为负值。 </li>
<li>10.SEQ read的序列 </li>
<li>11.<strong>QUAL</strong> read序列对应的ASCII码格式的碱基质量值 </li>
<li>12.可选的区域 header section</li>
</ul>
<h2 id="2-1-第1列：Qname"><a href="#2-1-第1列：Qname" class="headerlink" title="2.1 第1列：Qname"></a>2.1 第1列：Qname</h2><ul>
<li>Read的名字</li>
</ul>
<h2 id="2-2-第2列：FLAG"><a href="#2-2-第2列：FLAG" class="headerlink" title="2.2 第2列：FLAG"></a>2.2 第2列：FLAG</h2><ul>
<li><p>每一个read的比对情况可以用十进制数字（或者十六进制数字）表示，如果比对情况 有多个，将多个比对情况所代表的十进制数字加和就是这一行的FLAG。</p>
<ul>
<li>比如，图1中r001的FLAG是99（1+2+32+64），则表示了“该read是pair read中的一个”，“pair read中每个都能够正确比对上”，“该read的mate read的反向互补可以比对上”，“该read是pari read中的read1”；</li>
<li>r001的另一个FLAG是147（1+2+16+128），则表示“该read是pair read中的一个”，“pair read中每个都能够正确比对上”，“该read是原read的反向互补”，“该read是pari read中的read2”（也就是说，该read是read2的反向互补序列）。</li>
</ul>
</li>
<li><p>值得注意的是，r001是pair read，而且都能比对上，所以r001出现了两次，如果r001的read1比对到参考序列的2个地方，r001的名字则会出现三次；如果read1比对上一次，read2没有比对上，r001仍会出现2次，不过，其中一个r001的第三列为“*”；所以pair-end测序，read1文件和read2文件同时mapping，相同reads的id最少出现2次。</p>
</li>
<li><p>以下网站可以通过输入FLAG值，直接找出该FLAG是那些FLAG的加和：Decoding SAM flags（<a target="_blank" rel="noopener" href="https://broadinstitute.github.io/picard/explain-flags.html%EF%BC%89">https://broadinstitute.github.io/picard/explain-flags.html）</a></p>
</li>
</ul>
<table>
<thead>
<tr>
<th align="left">Value 10</th>
<th align="left">Value 16</th>
<th align="left">Description</th>
<th align="center">99</th>
<th align="center">83</th>
<th align="center">147</th>
<th align="center">163</th>
</tr>
</thead>
<tbody><tr>
<td align="left">1</td>
<td align="left">0x1</td>
<td align="left">read paired</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">2</td>
<td align="left">0x2</td>
<td align="left">read mapped in proper pair</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">4</td>
<td align="left">0x4</td>
<td align="left">read unmapped</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">8</td>
<td align="left">0x8</td>
<td align="left">mate unmapped</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">16</td>
<td align="left">0x10</td>
<td align="left">read reverse strand</td>
<td align="center"></td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center"></td>
</tr>
<tr>
<td align="left">32</td>
<td align="left">0x20</td>
<td align="left">mate reverse strand</td>
<td align="center">+</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">64</td>
<td align="left">0x40</td>
<td align="left">first in pair</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">128</td>
<td align="left">0x80</td>
<td align="left">second in pair</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">+</td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">256</td>
<td align="left">0x100</td>
<td align="left">not primary alignment</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">512</td>
<td align="left">0x200</td>
<td align="left">read fails platform&#x2F;vendor quality checks</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">1024</td>
<td align="left">0x400</td>
<td align="left">read is PCR or optical duplicate</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">2048</td>
<td align="left">0x800</td>
<td align="left">supplementary alignment</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
</tbody></table>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">-------------------------------------------------------------</span><br><span class="line">---R1---&gt; 99                  --------R2------&gt;  163</span><br><span class="line"></span><br><span class="line">          &lt;----R2----  147            &lt;--------R1------  83</span><br><span class="line">------------------------------------------------------------- reverse strand</span><br></pre></td></tr></table></figure>

<h2 id="2-3-第3列：RNAME"><a href="#2-3-第3列：RNAME" class="headerlink" title="2.3 第3列：RNAME"></a>2.3 第3列：RNAME</h2><ul>
<li>比对上的参考序列的名字，该名字出现在Header section的@SQ行的SN标识中，如果该read没有比对上，也就是说该read在参考序列上没有坐标，那么这一列则用“”表示，那么这一行的POS和CIGAR列也会是“”。</li>
</ul>
<h2 id="2-4-第4列：POS"><a href="#2-4-第4列：POS" class="headerlink" title="2.4 第4列：POS"></a>2.4 第4列：POS</h2><ul>
<li>read比对到的参考序列“RNAME”最左侧的位置坐标，也是CIGAR中第一个比对标识“M”对应的最左侧碱基在参考序列的位置，未比对上的read在参考序列中没有坐标，此列标识为“0”。</li>
</ul>
<h2 id="2-5-第5列：MAPQ"><a href="#2-5-第5列：MAPQ" class="headerlink" title="2.5 第5列：MAPQ"></a>2.5 第5列：MAPQ</h2><ul>
<li>比对的质量值，计算方法为比对错误率的-10*log10的值，一般是四舍五入的整数值，如果是255，说明该比对值无效。</li>
</ul>
<h2 id="2-6-第6列：CIGAR"><a href="#2-6-第6列：CIGAR" class="headerlink" title="2.6 第6列：CIGAR"></a>2.6 第6列：CIGAR</h2><p>CIGAR标识符表示read中每个碱基的比对情况，主要有以下标识符：</p>
<ul>
<li><p>M: alignment match (can be a sequence match or mismatch)<br>read上的碱基与参考序列“RNAME”完全匹配，碱基一一对应，包括了正确匹配与错误匹配</p>
</li>
<li><p>I: insertion to the reference<br>read上的碱基相对于参考序列“RNAME”有插入现象（如下）：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">REF:  CACGATCA**GACCGATACGTCCGA </span><br><span class="line">READ1:  CGATCAGAGACCGATA </span><br><span class="line">CIGAR：6M2I8M</span><br></pre></td></tr></table></figure></li>
<li><p>D: deletion from the reference<br>read上的碱基相对于参考序列“RNAME”有删除现象（如下）：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">REF: AGCTAGCATCGTGTCGCCCGTCTAGCATACGC</span><br><span class="line">READ:             TCGCCCGT-TAGCAT</span><br><span class="line">CIGAR：8M1D6M</span><br></pre></td></tr></table></figure></li>
<li><p>N: skipped region from the reference<br>read上的碱基相对于参考序列“RNAME”存在连续没有比对上的空缺，这些空缺用N来表示，跟“D”相似但远比“D”缺失的更多，这种比对类型也叫“Spliced alignment”，常见cDNA与参考序列比对（如下）:”…”表示intron</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">REF:AGCATCGTGTCGCCCGTCTAGCATACGCATGATCGACTGTCAGCTAGTCAGACTA</span><br><span class="line">READ:     GTGTAACCC................................TCAGAATA</span><br><span class="line">CIGAR：9M32N8M</span><br></pre></td></tr></table></figure></li>
<li><p>S: soft clipping (clipped sequences present in SEQ)</p>
</li>
<li><p>H: hard clipping (clipped sequences NOT present in SEQ)<br>read的开头或者结尾部分没有比对到参考序列”RNAME”上，但这部分未比对上的连续序列仍保留在sam文件的该read序列中，用“S”来表示；如果未保留，则用“H”表示，也即“hard cliping”（如下所示，也可同图2中r003的比对CIGAR中看出）</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">REF: AGCTAGCATCGTGTCGCCCGTCTAGCATACGCAT</span><br><span class="line">READ:          gggTCGCCCGT-TAGCATgggg</span><br><span class="line">CIGAR：3S8M1D6M4S （在sam中存储为GGGGTGTAACCGACTAGGGGG）</span><br><span class="line">CIGAR：3H8M1D6M4H （在sam中存储为GTGTAACCGACTAG）</span><br></pre></td></tr></table></figure></li>
<li><p>P: padding (silent deletion from padded reference)<br>多条read比对到参考序列的同一位置时，如果不同read单独同该参考序列比对时，参考序列的情况也不同，比如下方READ1同参考序列比对时，“GA”属于插入（6M2I8M），READ2同参考序列比对时，“A”属于插入（4M1I9M ），READ3同参考序列完全匹配（10M），没有插入，但是三条read之前却没有可比性。因此，当参考序列“比对情况包含完整”且序列唯一时，所有read同时进行比对，read3这种原本没有插入却默认插入的比对称之为Padded alignment，这种情况用“P”表示。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">REF:  CACGATCA**GACCGATACGTCCGA</span><br><span class="line">READ1:  CGATCAGAGACCGATA </span><br><span class="line">READ2:    ATCA*AGACCGATAC </span><br><span class="line">READ3:   GATCA**GACCG  </span><br><span class="line">The padded CIGAR are different: </span><br><span class="line">READ1: 6M2I8M </span><br><span class="line">READ2: 4M1P1I9M  </span><br><span class="line">READ3: 5M2P5M</span><br></pre></td></tr></table></figure></li>
<li><p>&#x3D;：sequence match 正确匹配</p>
</li>
<li><p>X：sequence mismatch 错误匹配</p>
</li>
</ul>
<blockquote>
<p>perl CIGAR代码 为了检测每个read的甲基化情况</p>
</blockquote>
  <figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">my</span> @f=<span class="keyword">split</span>(<span class="regexp">/\s+/</span>,$line);</span><br><span class="line"></span><br><span class="line"><span class="keyword">my</span> $flag = $f[<span class="number">1</span>];</span><br><span class="line"><span class="keyword">my</span> $chr = $f[<span class="number">2</span>];</span><br><span class="line"><span class="keyword">my</span> $pos = $f[<span class="number">3</span>];</span><br><span class="line"><span class="keyword">my</span> $cigar_2 = $f[<span class="number">5</span>];</span><br><span class="line"></span><br><span class="line"><span class="comment"># parsing CIGAR string</span></span><br><span class="line"><span class="keyword">my</span> @len = <span class="keyword">split</span> (<span class="regexp">/\D+/</span>,$cigar_2); <span class="comment"># storing the length per operation</span></span><br><span class="line"><span class="keyword">my</span> @ops = <span class="keyword">split</span> (<span class="regexp">/\d+/</span>,$cigar_2); <span class="comment"># storing the operation</span></span><br><span class="line"><span class="keyword">shift</span> @ops; <span class="comment"># remove the empty first element</span></span><br><span class="line"><span class="keyword">die</span> <span class="string">&quot;CIGAR string contained a non-matching number of lengths and operations ($cigar_2)\n&quot;</span> <span class="keyword">unless</span> (<span class="keyword">scalar</span> @len == <span class="keyword">scalar</span> @ops);</span><br><span class="line"></span><br><span class="line"><span class="comment">### determining end position of the read</span></span><br><span class="line"><span class="keyword">my</span> $readPos = <span class="number">0</span>;</span><br><span class="line"><span class="keyword">my</span> $genomePos = $pos;</span><br><span class="line"></span><br><span class="line"><span class="keyword">foreach</span> <span class="keyword">my</span> $index(<span class="number">0</span>..$#len)&#123;</span><br><span class="line"></span><br><span class="line">	<span class="keyword">my</span> $lengthSeg = $len[$index];</span><br><span class="line"></span><br><span class="line">	<span class="keyword">if</span> ($ops[$index] eq <span class="string">&#x27;M&#x27;</span>)&#123;  <span class="comment"># standard matching bases</span></span><br><span class="line"></span><br><span class="line">		$readPos += $lengthSeg;</span><br><span class="line">		$genomePos += $lengthSeg;</span><br><span class="line">	  </span><br><span class="line">	&#125;</span><br><span class="line">	<span class="keyword">elsif</span>($ops[$index] eq <span class="string">&#x27;I&#x27;</span>)&#123; <span class="comment"># insertions do not affect the end position</span></span><br><span class="line">	  </span><br><span class="line">		$readPos += $lengthSeg;</span><br><span class="line"></span><br><span class="line">	&#125;</span><br><span class="line">	<span class="keyword">elsif</span>($ops[$index] eq <span class="string">&#x27;D&#x27;</span>)&#123; <span class="comment"># deletions do affect the end position</span></span><br><span class="line">	  $genomePos += $lengthSeg;</span><br><span class="line"></span><br><span class="line">	&#125;<span class="keyword">elsif</span>($ops[$index] eq <span class="string">&#x27;S&#x27;</span>)&#123; <span class="comment"># deletions do affect the end position</span></span><br><span class="line">	    $readPos += $lengthSeg;</span><br><span class="line">	&#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br></pre></td></tr></table></figure>


<h2 id="2-7-第7列：RNEXT"><a href="#2-7-第7列：RNEXT" class="headerlink" title="2.7 第7列：RNEXT"></a>2.7 第7列：RNEXT</h2><ul>
<li><p>该read的mate read比对上的参考序列的名字，该名字出现在Header section的@SQ行的SN标识中，</p>
</li>
<li><p>如果和该read所在行的第三列“RNAME”一样，则用“&#x3D;”表示，说明这对read比对到了同一条参考序列上；</p>
</li>
<li><p>如果mate read没有比对上，第七列则用“*”表示；</p>
</li>
<li><p>如果这对read没有比对到同一条参考序列，那么这一列则是mate read所在行第三列的“RNAME”。</p>
</li>
</ul>
<h2 id="2-8-第8列：PNEXT"><a href="#2-8-第8列：PNEXT" class="headerlink" title="2.8 第8列：PNEXT"></a>2.8 第8列：PNEXT</h2><ul>
<li>该read的mate read比对到的参考序列“RNAME”<strong>最左侧</strong>的位置坐标，也是mate read CIGAR中第一个比对标识“M”对应的最左侧碱基在参考序列的位置，未比对上的read在参考序列中没有坐标，此列标识为“0”。</li>
</ul>
<h2 id="2-9-第9列：ISIZE"><a href="#2-9-第9列：ISIZE" class="headerlink" title="2.9 第9列：ISIZE"></a>2.9 第9列：ISIZE</h2><ul>
<li><p>表示pair read完全匹配到同一条参考序列时，两个read之间的长度，可简单理解为测序文库的长度。这个定义有两种情况（虚线表示未比对上的序列，即soft-clipped bases）：</p>
</li>
<li><p>第一种情况如图左所示，Read1和Read2比对到同一条参考序列，此时ISIZE即为Read2的最右侧比对坐标减去Read1最左侧比对坐标；</p>
</li>
<li><p>第二种情况如图右所示，由于至今没有明确的定义和共识，因此ISIZE可以是TLEN#1，也可以是TLEN#2，视情况而定。</p>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">                                            |&lt;--TLEN#1----&gt;|</span><br><span class="line">-------------------------------   -----------------------------------</span><br><span class="line">|---R1---&gt; 99                               |---R1-----------------&gt; 99 </span><br><span class="line"></span><br><span class="line">                &lt;----R2----| 147     &lt;----------R2---------|  147</span><br><span class="line">-------------------------------   ------------------------------------</span><br><span class="line">|&lt;-------TLEN #1/#2-------&gt;|         |&lt;-------TLEN #2--------------&gt;|</span><br></pre></td></tr></table></figure>

<h2 id="2-10-第10列：SEQ"><a href="#2-10-第10列：SEQ" class="headerlink" title="2.10 第10列：SEQ"></a>2.10 第10列：SEQ</h2><ul>
<li>存储的序列，没有存储，此列则用“*”标识。该序列的长度<strong>一定等于</strong>CIGAR标识中“M”，“I”，“S”，“&#x3D;”，“X”标识的碱基长度之和。</li>
</ul>
<h2 id="2-11-第11列：QUAL-Base-quality-score"><a href="#2-11-第11列：QUAL-Base-quality-score" class="headerlink" title="2.11 第11列：QUAL:Base quality score"></a>2.11 第11列：QUAL:Base quality score</h2><ul>
<li><p>序列的每个碱基对应一个碱基质量字符，每个碱基质量字符对应的ASCII码值减去33（Sanger Phred-33 质量值体系），即为该碱基的测序质量得分（Phred Quality Score）。不同Phred Quality Score代表不同的碱基测序错误率，如Phred Quality Score值为20和30分别表示碱基测序错误率为1%和0.1%。（ASCII码从0-127，为什么很多Phred评分从33开始呢？因为33是第一个肉眼可见的字符!，在此之前的都是控制符，比如空格、制表等，不方便用这些控制符来表示一个碱基的质量。）</p>
</li>
<li><p>假设一个碱基测序错误的概率是P，那么此时对应的Phred quality score（常用Q表示）</p>
<blockquote>
<p>$Q &#x3D; -10log_{10}(P)$</p>
</blockquote>
</li>
<li><p>假设100个碱基中有1个出现错误，即P&#x3D;0.01，根据上式可以计算，此时Q&#x3D;20；同理，当测序错误率为0.001时，此时Q&#x3D;30。<br>测序结果中的每一个碱基都有这么一个Phred评分。为了便于描述，人们习惯上将Phred评分对应一个ASCII码，这样每一个碱基的质量都可以用一个字符来表述了。</p>
</li>
</ul>
<table>
<thead>
<tr>
<th align="left">P</th>
<th align="center">Q (Phred quality score)</th>
<th align="center">Sanger Phred</th>
<th align="left">ASCII</th>
</tr>
</thead>
<tbody><tr>
<td align="left">0.01</td>
<td align="center">20</td>
<td align="center">53</td>
<td align="left"><strong>5</strong></td>
</tr>
<tr>
<td align="left">0.001</td>
<td align="center">30</td>
<td align="center">63</td>
<td align="left"><strong>?</strong></td>
</tr>
<tr>
<td align="left">0.0001</td>
<td align="center">40</td>
<td align="center">73</td>
<td align="left"><strong>I</strong></td>
</tr>
<tr>
<td align="left">0.00001</td>
<td align="center">50</td>
<td align="center">83</td>
<td align="left"><strong>S</strong></td>
</tr>
</tbody></table>
<h1 id="3-samtools-软件使用"><a href="#3-samtools-软件使用" class="headerlink" title="3. samtools 软件使用"></a>3. samtools 软件使用</h1><h2 id="3-1-sort：-sort-alignment-file"><a href="#3-1-sort：-sort-alignment-file" class="headerlink" title="3.1 sort： sort alignment file"></a>3.1 sort： sort alignment file</h2><ul>
<li>本命令对bam文件中的序列进行排序，默认下是按序列在fasta文件中的顺序（即header）和序列从左往右的位点排序。</li>
</ul>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">samtools <span class="keyword">sort</span> [-T out.prefix][-@ threads][-m maxMem][-o out.sorted.bam][in.bam]</span><br><span class="line">samtools <span class="keyword">sort</span> -@ <span class="number">10</span> -o abc.sorted.bam abc.bam</span><br><span class="line">samtools <span class="keyword">sort</span> -m <span class="number">500000000</span> -o abc.sorted.bam abc.bam <span class="comment"># 500M dang abc.bam文件比较大，split的文件比较多，比如200多个。</span></span><br></pre></td></tr></table></figure>
<h2 id="3-2-merge：-merge-sorted-alignments"><a href="#3-2-merge：-merge-sorted-alignments" class="headerlink" title="3.2 merge： merge sorted alignments"></a>3.2 merge： merge sorted alignments</h2><ul>
<li><p>本命令将多个排好序的bam比对文件进行合并，产生一个排好序的bam输出文件(合并后的文件不需要再进行sort)，这个文件含有所有的输入记录，并且保留了他们原来的顺序。</p>
<blockquote>
<p>samtools merge [out.bam][in1.bam][in2.bam][in3.bam]</p>
</blockquote>
</li>
</ul>
<h2 id="3-3-index：index-alignment"><a href="#3-3-index：index-alignment" class="headerlink" title="3.3 index：index alignment"></a>3.3 index：index alignment</h2><ul>
<li><p>本命令对bam文件建立索引并产生后缀为.bai的文件，用于快速的随机处理。很多后续分析的过程需要有bai文件的存在，特别是显示序列比对情况下，比如samtool的tview命令等。#必须对bam文件进行默认情况下的排序后，才能进行index。否则会报错</p>
<blockquote>
<p>samtools index aln.sorted.bam</p>
</blockquote>
</li>
</ul>
<h2 id="3-4-view：SAMBAMCRAM-conversion"><a href="#3-4-view：SAMBAMCRAM-conversion" class="headerlink" title="3.4 view：SAM&lt;-&gt;BAM&lt;-&gt;CRAM conversion"></a>3.4 view：SAM&lt;-&gt;BAM&lt;-&gt;CRAM conversion</h2><ul>
<li>本命令将sam文件转换成bam文件；然后对bam文件进行各种操作，比如数据的排序和提取(这些操作是对bam文件进行的，不能对sam文件进行该操作)；最后将排序或提取得到的数据输出为bam或sam格式。bam文件优点：bam文件为二进制文件，占用的磁盘空间比sam文本文件小；利用bam二进制文件的运算速度快。</li>
</ul>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">samtools view -b abc.sam &gt; abc.bam</span><br><span class="line">samtools view -b abc.sam -o abc.bam</span><br></pre></td></tr></table></figure>

<h2 id="3-5-tview"><a href="#3-5-tview" class="headerlink" title="3.5 tview"></a>3.5 tview</h2><ul>
<li>直观显示reads比对到基因组的情况，与IGV类似需要先sort和index</li>
</ul>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools tview abc.sorted.bam hg19.fasta</span><br></pre></td></tr></table></figure>
<ul>
<li>press <strong>g</strong> to check the alignment start from a region in the format like chr10:10,000,000 or &#x3D;10,000,000 when viewing the same reference sequence.</li>
<li>c, C, n …</li>
</ul>
<h2 id="3-6-depth"><a href="#3-6-depth" class="headerlink" title="3.6 depth"></a>3.6 depth</h2><ul>
<li>统计每个碱基位点的测序深度;需要使用重定向定义输出文件；要使用构建过索引的bam Usage: bam2depth [-r reg] [-q baseQthres] [-Q mapQthres] [-b in.bed] […] -r：（region）加染色体号； -q：要求测序碱基质量最低值； -Q：要求比对的质量最低值<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools depth abc.sorted.bam &gt;abc.depth</span><br></pre></td></tr></table></figure></li>
</ul>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2024/01/22/bam%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/" data-id="clw7l0lmu0006olgpa2x51o7r" data-title="Sam/Bam文件格式详解" class="article-share-link"><span class="fa fa-share">Share</span></a>
      
      
      
    </footer>
  </div>
  
</article>



  
    <article id="post-2.常见软件安装整理" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2024/01/22/2.%E5%B8%B8%E8%A7%81%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85%E6%95%B4%E7%90%86/" class="article-date">
  <time class="dt-published" datetime="2024-01-22T07:08:14.000Z" itemprop="datePublished">2024-01-22</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2024/01/22/2.%E5%B8%B8%E8%A7%81%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85%E6%95%B4%E7%90%86/">常见生信软件安装</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <h1 id="常见生信软件安装"><a href="#常见生信软件安装" class="headerlink" title="常见生信软件安装"></a>常见生信软件安装</h1><h2 id="1-手动安装"><a href="#1-手动安装" class="headerlink" title="1. 手动安装"></a>1. 手动安装</h2><p>bwa, samtools, bcftools, bgzip, tabix</p>
<h3 id="1-bwa"><a href="#1-bwa" class="headerlink" title="1.bwa"></a>1.bwa</h3><h3 id="2-samtools-bcftools-版本要一致-1-17"><a href="#2-samtools-bcftools-版本要一致-1-17" class="headerlink" title="2.samtools, bcftools 版本要一致 #1.17"></a>2.samtools, bcftools 版本要一致 #1.17</h3><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">#Building each desired package from source is very simple:</span><br><span class="line">    cd samtools-1.x    # and similarly for bcftools and htslib</span><br><span class="line">    ./configure --prefix=/where/to/install</span><br><span class="line">    make</span><br><span class="line">    make install</span><br><span class="line"></span><br><span class="line">    ### install bgzip, tabix</span><br><span class="line">    cd .../samtools-1.17 # Within the unpacked release directory</span><br><span class="line">    ./configure --prefix=/path/to/location</span><br><span class="line">    make all all-htslib</span><br><span class="line">    make install install-htslib</span><br></pre></td></tr></table></figure>

<h2 id="2-conda安装包安装"><a href="#2-conda安装包安装" class="headerlink" title="2. conda安装包安装"></a>2. conda安装包安装</h2><h3 id="1-conda-install-c-bioconda-pysam-会安装一下3个包"><a href="#1-conda-install-c-bioconda-pysam-会安装一下3个包" class="headerlink" title="1.conda install -c bioconda pysam #会安装一下3个包"></a>1.conda install -c bioconda pysam #会安装一下3个包</h3><pre><code>samtools=1.3.1
bcftools=1.3.1
htslib=1.3.1
</code></pre>
<h3 id="2-snakemake"><a href="#2-snakemake" class="headerlink" title="2.snakemake"></a>2.snakemake</h3><pre><code>安装 mamba
conda install -n base -c conda-forge mamba
将 snakemake 安装到虚拟环境
mamba create -c conda-forge -c bioconda -n snakemake snakemake

conda activate snakemake
snakemake --help

git clone https://bitbucket.org/snakemake/snakemake-tutorial.git
cd snakemake-tutorial
tree

安装pipeline所需要的包，比如bwa，samtools等
conda install -c conda-forge graphviz #dot画图 2.42.3 装在 snakemake 的环境下面。

snakemake --dag  | dot -Tsvg &gt; dag.svg #生成有向无环图 (DAG)
snakemake --cores 1 
snakemake --cores 1
snakemake -s Snakefile
</code></pre>
<h3 id="3"><a href="#3" class="headerlink" title="3"></a>3</h3><h2 id="3-R-安装包"><a href="#3-R-安装包" class="headerlink" title="3.R 安装包"></a>3.R 安装包</h2><p>conda active r-base<br>conda install R </p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2024/01/22/2.%E5%B8%B8%E8%A7%81%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85%E6%95%B4%E7%90%86/" data-id="clw7l0lmr0002olgp50gvhify" data-title="常见生信软件安装" class="article-share-link"><span class="fa fa-share">Share</span></a>
      
      
      
    </footer>
  </div>
  
</article>



  
    <article id="post-1.linuxCommandLine" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2024/01/22/1.linuxCommandLine/" class="article-date">
  <time class="dt-published" datetime="2024-01-22T06:08:14.000Z" itemprop="datePublished">2024-01-22</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2024/01/22/1.linuxCommandLine/">Linux 常见命令</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <h1 id="Unix-Linux-操作系统介绍"><a href="#Unix-Linux-操作系统介绍" class="headerlink" title="Unix&#x2F;Linux 操作系统介绍"></a>Unix&#x2F;Linux 操作系统介绍</h1><h2 id="1-文件的cp、rm和mv命令"><a href="#1-文件的cp、rm和mv命令" class="headerlink" title="1. 文件的cp、rm和mv命令"></a>1. 文件的cp、rm和mv命令</h2><h3 id="cp-复制"><a href="#cp-复制" class="headerlink" title="cp 复制"></a>cp 复制</h3><ul>
<li>说明:该命令的功能是将指定的文件或目录拷贝到另一文件或目录中。可以使用通配符拷贝具 有同一特征的所有文件。</li>
<li>语法: cp [参数] 源文件或目录 目标文件或目录</li>
<li>参数:<ul>
<li>-a 该选项通常在拷贝目录时使用。它保留链接、文件属性，并递归地拷贝目录，其作用等于dpR 选项的组合。</li>
<li>-d 拷贝时保留链接。</li>
<li>-f 删除已经存在的目标文件而不提示。</li>
<li>-i 和 f 选项相反，在覆盖目标文件之前将给出提示要求用户确认。回答 y 时目标文件将被覆盖，是交互式拷贝。为防止用户在不经意的情况下用 cp 命令破坏另一个文件，如用户指定 的目标文件名已存在，用 cp 命令拷贝文件后，这个文件就会被新源文件覆盖，因此，建议 用户在使用 cp 命令拷贝文件时，最好使用 i 选项。</li>
<li>-p 此时 cp 除复制源文件的内容外，还将把其修改时间和访问权限也复制到新文件中。</li>
<li>-r 若给出的源文件是一目录文件，此时 cp 将递归复制该目录下所有的子目录和文件。此时目标文件必须为一个目录名。 </li>
<li>-l 不作拷贝，只是链接文件。
        
          <p class="article-more-link">
            <a href="/2024/01/22/1.linuxCommandLine/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2024/01/22/1.linuxCommandLine/" data-id="clw7l0lmp0001olgpfbq42ry1" data-title="Linux 常见命令" class="article-share-link"><span class="fa fa-share">Share</span></a>
      
      
      
    </footer>
  </div>
  
</article>



  


</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Categories</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8F/">二代测序</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a></li></ul>
    </div>
  </div>


  
    

  
    
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2024/01/">January 2024</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2024/01/23/GATK4%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95%E8%AF%A6%E8%A7%A3-%E5%8F%98%E5%BC%82%E6%A3%80%E6%B5%8B/">GATK使用方法详解-变异检测</a>
          </li>
        
          <li>
            <a href="/2024/01/22/VCF%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/">VCF文件格式详解</a>
          </li>
        
          <li>
            <a href="/2024/01/22/bam%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F/">Sam/Bam文件格式详解</a>
          </li>
        
          <li>
            <a href="/2024/01/22/2.%E5%B8%B8%E8%A7%81%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85%E6%95%B4%E7%90%86/">常见生信软件安装</a>
          </li>
        
          <li>
            <a href="/2024/01/22/1.linuxCommandLine/">Linux 常见命令</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2024 John Doe<br>
      Powered by <a href="https://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    


<script src="/js/jquery-3.6.4.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>