

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/fluid.png">
  <link rel="icon" href="/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="Qiang">
  <meta name="keywords" content="">
  
    <meta name="description" content="Sam&#x2F;Bam文件格式详解bam文件是sam文件的二进制格式，具有更小的存储空间。并且许多下游分析工具使用的是BAM格式。格式详见说明手册(https:&#x2F;&#x2F;samtools.github.io&#x2F;hts-specs&#x2F;SAMv1.pdf) bam文件格式是生物信息最常用的文件格式，是短序列比对默认的标准格式，以TAB为分割符的文本格式。一般测序下机数据的mapping结果，均以该格式进行存储">
<meta property="og:type" content="article">
<meta property="og:title" content="4.Sam&#x2F;Bam文件格式详解">
<meta property="og:url" content="https://xiaoqiangq.github.io/2024/01/25/4.bamFormat/index.html">
<meta property="og:site_name" content="Qiang Wei&#39;s Lab">
<meta property="og:description" content="Sam&#x2F;Bam文件格式详解bam文件是sam文件的二进制格式，具有更小的存储空间。并且许多下游分析工具使用的是BAM格式。格式详见说明手册(https:&#x2F;&#x2F;samtools.github.io&#x2F;hts-specs&#x2F;SAMv1.pdf) bam文件格式是生物信息最常用的文件格式，是短序列比对默认的标准格式，以TAB为分割符的文本格式。一般测序下机数据的mapping结果，均以该格式进行存储">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://xiaoqiangq.github.io/2024/01/25/4.bamFormat/sam_format_annotated_example.jpg">
<meta property="og:image" content="https://xiaoqiangq.github.io/2024/01/25/4.bamFormat/sam_flag.png">
<meta property="article:published_time" content="2024-01-25T08:08:14.000Z">
<meta property="article:modified_time" content="2024-05-24T07:44:33.002Z">
<meta property="article:author" content="Qiang Wei">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://xiaoqiangq.github.io/2024/01/25/4.bamFormat/sam_format_annotated_example.jpg">
  
  
    <meta name="referrer" content="no-referrer-when-downgrade">
  
  
  <title>4.Sam/Bam文件格式详解 - Qiang Wei&#39;s Lab</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"xiaoqiangq.github.io","root":"/","version":"1.9.7","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"left","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":true,"follow_dnt":true,"baidu":null,"google":{"measurement_id":null},"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null,"path":"window.location.pathname","ignore_local":false}},"search_path":"/local-search.xml","include_content_in_search":true};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
  

  

  
    <!-- Google tag (gtag.js) -->
    <script async>
      if (!Fluid.ctx.dnt) {
        Fluid.utils.createScript("https://www.googletagmanager.com/gtag/js?id=", function() {
          window.dataLayer = window.dataLayer || [];
          function gtag() {
            dataLayer.push(arguments);
          }
          gtag('js', new Date());
          gtag('config', '');
        });
      }
    </script>
  

  

  

  

  



  
<meta name="generator" content="Hexo 7.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 30vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/">
      <strong>Qiang</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/" target="_self">
                <i class="iconfont icon-home-fill"></i>
                <span>首页</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/" target="_self">
                <i class="iconfont icon-archive-fill"></i>
                <span>归档</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/" target="_self">
                <i class="iconfont icon-category-fill"></i>
                <span>分类</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/" target="_self">
                <i class="iconfont icon-tags-fill"></i>
                <span>标签</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/" target="_self">
                <i class="iconfont icon-user-fill"></i>
                <span>关于</span>
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              <i class="iconfont icon-search"></i>
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">
              <i class="iconfont icon-dark" id="color-toggle-icon"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="4.Sam/Bam文件格式详解"></span>
          
        </div>

        
          
  <div class="mt-3">
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2024-01-25 16:08" pubdate>
          2024年1月25日 下午
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          4.1k 字
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          34 分钟
        
      </span>
    

    
    
      
        <span id="busuanzi_container_page_pv" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="busuanzi_value_page_pv"></span> 次
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="padding-left: 2rem; margin-right: -1rem">
    <div id="toc">
  <p class="toc-header">
    <i class="iconfont icon-list"></i>
    <span>目录</span>
  </p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <h1 id="seo-header">4.Sam/Bam文件格式详解</h1>
            
              <p id="updated-time" class="note note-info" style="">
                
                  
                    本文最后更新于 2024年5月24日 下午
                  
                
              </p>
            
            
              <div class="markdown-body">
                
                <h1 id="Sam-Bam文件格式详解"><a href="#Sam-Bam文件格式详解" class="headerlink" title="Sam&#x2F;Bam文件格式详解"></a>Sam&#x2F;Bam文件格式详解</h1><p>bam文件是sam文件的二进制格式，具有更小的存储空间。并且许多下游分析工具使用的是BAM格式。格式详见说明手册(<a target="_blank" rel="noopener" href="https://samtools.github.io/hts-specs/SAMv1.pdf">https://samtools.github.io/hts-specs/SAMv1.pdf</a>)</p>
<p>bam文件格式是生物信息最常用的文件格式，是短序列比对默认的标准格式，以TAB为分割符的文本格式。一般测序下机数据的mapping结果，均以该格式进行存储。而用于序列比对的工具，如bwa、bowtie、bismark、blast都提供了bam格式的输出形式。因此可以说，对生物信息学工具的优化和开发，几乎不可避免地需要对bam处理，对于基础用户而言，可能经常用到samtools view .bam | less -SN可以查看二进制的bam文件。</p>
<p>bam格式包含标头部分（header section）和比对部分（alignment section）见下图</p>
<img src="/2024/01/25/4.bamFormat/sam_format_annotated_example.jpg" srcset="/img/loading.gif" lazyload class="">

<h2 id="1-Header-section"><a href="#1-Header-section" class="headerlink" title="1. Header section"></a>1. Header section</h2><ul>
<li>该部分全部以“@”开头，后紧跟的两个大写字母主要有HD，SQ，RG，PG和CO五类。涵盖了文件标准格式版本（VN）、比对中使用的参考序列信息（SQ）、测序数据分组信息（RG）、比对或后期处理使用的程序信息（PG）等。不仅如此，用户也可以根据后期分析需要自定义一系列自定义的信息，最为常用的就是UMI标签信息。</li>
<li>@HD 这一行中有各种不同的标识<ul>
<li>标识“VN”用以说明SAM格式版本</li>
<li>标识“SO”用以说明比对排序的情况，有unknown (default)、unsorted、queryname和coordinate,对于coordinate，排序的主键是Alignments section的第三列“RNAME”，其顺序由@SQ行的“SN”标识的顺序定义，次要排序键是Alignments section的第四列“POS”字段。对于RNAME和POS相等的比对，排列顺序则是任意的</li>
</ul>
</li>
<li>@SQ reference染色体信息，它的值主要是用于Alignments section的第三列“RNAME”和第七列“MRNM”比对的记录<ul>
<li>SN:Chr1则表示参考序列名称</li>
<li>LN:100参考序列长度</li>
</ul>
</li>
<li>@RG read序列信息</li>
<li>@PG 使用的程序说明；该行“ID”为程序记录标识符<ul>
<li>PN为程序名字</li>
<li>CL为命令行</li>
</ul>
</li>
<li>@CO行是任意的说明信息</li>
</ul>
<p><strong>注意事项</strong><br>header里面包含如下重要信息</p>
<ol>
<li><strong>mapping(bwa,GATK,bismark,bowtie2,…)</strong></li>
<li><strong>duplication(picard,samblaster,sambamba,…)</strong></li>
<li><strong>reference genome(hg19,hg38)</strong></li>
</ol>
<h2 id="2-Alignments-section"><a href="#2-Alignments-section" class="headerlink" title="2. Alignments section"></a>2. Alignments section</h2><p>该部分是SAM文件的核心部分，每一行代表一个序列的线性比对（linear alignment of a segment），每行包含前<strong>11</strong>个必需字段，和第12个字段后多个可选字段，使用TAB-separated分割，当某个字段信息缺省时，如果字段是字符串型以*替代，如果字段是整型以’0’来替代，下表为11个必需字段含义的概述。</p>
<table>
<thead>
<tr>
<th align="left">Col</th>
<th align="left">Field</th>
<th align="center">Type</th>
<th align="center">Regexp&#x2F;Range</th>
<th align="left">Brief description</th>
</tr>
</thead>
<tbody><tr>
<td align="left">1</td>
<td align="left">QNAME</td>
<td align="center">String</td>
<td align="center">[!-?A-~]{1,254}</td>
<td align="left">Query template NAME</td>
</tr>
<tr>
<td align="left">2</td>
<td align="left">FLAG</td>
<td align="center">Int</td>
<td align="center">[0,2^16−1]</td>
<td align="left">bitwise FLAG</td>
</tr>
<tr>
<td align="left">3</td>
<td align="left">RNAME</td>
<td align="center">String</td>
<td align="center">\*|[:rname:∧*&#x3D;][:rname:]*</td>
<td align="left">Reference sequence NAME 1-based leftmost mapping</td>
</tr>
<tr>
<td align="left">4</td>
<td align="left">POS</td>
<td align="center">Int</td>
<td align="center">[0,2^31−1]</td>
<td align="left">POSition</td>
</tr>
<tr>
<td align="left">5</td>
<td align="left">MAPQ</td>
<td align="center">Int</td>
<td align="center">[0,2^8−1]</td>
<td align="left">MAPping Quality</td>
</tr>
<tr>
<td align="left">6</td>
<td align="left">CIGAR</td>
<td align="center">String</td>
<td align="center">\*|([0-9]+[MIDNSHPX&#x3D;])+</td>
<td align="left">CIGAR string</td>
</tr>
<tr>
<td align="left">7</td>
<td align="left">RNEXT</td>
<td align="center">String</td>
<td align="center">\*|&#x3D;|[:rname:^*&#x3D;][:rname:]*</td>
<td align="left">Reference name of the mate&#x2F;next read</td>
</tr>
<tr>
<td align="left">8</td>
<td align="left">PNEXT</td>
<td align="center">Int</td>
<td align="center">[0,2^31−1]</td>
<td align="left">Position of the mate&#x2F;next read</td>
</tr>
<tr>
<td align="left">9</td>
<td align="left">TLEN</td>
<td align="center">Int</td>
<td align="center">[−2^31+1,2^31−1]</td>
<td align="left">observed Template Length</td>
</tr>
<tr>
<td align="left">10</td>
<td align="left">SEQ</td>
<td align="center">String</td>
<td align="center">\*|[A-Za-z&#x3D;.]+</td>
<td align="left">segment SEQuence</td>
</tr>
<tr>
<td align="left">11</td>
<td align="left">QUAL</td>
<td align="center">String</td>
<td align="center">[!-~]+</td>
<td align="left">ASCII of Phred-scaled base QUALity+33</td>
</tr>
</tbody></table>
<ul>
<li>1.QNAME序列的名字，也就是reads的名称</li>
<li>2.<strong>FLAG</strong>是一个标记的数字，描述该reads在参考基因的比对情况, 该FLAG是一串2进制代码</li>
<li>3.RNAME参考序列的名字</li>
<li>4.POS在参考序列上的位置</li>
<li>5.MAPQ mapping qulity越高则位点越独特，比对的质量值</li>
<li>6.<strong>CIGAR</strong>代表比对结果的CIGAR字符串，如37M1D2M1I，这段字符的意思是37个匹配，1个参考序列上的删除，2个匹配，1个参考序列上的插入。M代表的是alignment match(可以是错配)，可以理解为表示比对的具体情况</li>
<li>7.RNEXT mate序列所在参考序列的名称，mate一般指大的片段序列</li>
<li>8.PNEXT mate序列在参考序列上的位置</li>
<li>9.<strong>TLEN</strong>估计出的片段的长度，当mate序列位于本序列上游时该值为负值</li>
<li>10.SEQ read的序列</li>
<li>11.<strong>QUAL</strong>read序列对应的ASCII码格式的碱基质量值</li>
<li>12.可选的区域header section。值得注意的是，追加信息标签（TAG）的标准定义是一直在不断更新的，基本上每年都会有新增标签加入。</li>
</ul>
<h3 id="1-Qname"><a href="#1-Qname" class="headerlink" title="1.Qname"></a>1.Qname</h3><ul>
<li>比对read 的name（query template name），如果QNAME唯一，则序列被认为来源于同一模板；’*’表示该字段缺省；一般情况下，该字段为FASTQ文件的第一行信息；嵌合（Chimeric alignment）比对或者多次比对（Multiple mapping）的序列会导致一个QNAME在SAM中多次出现。</li>
</ul>
<h3 id="2-FLAG"><a href="#2-FLAG" class="headerlink" title="2.FLAG"></a>2.FLAG</h3><ul>
<li>每一个read的比对情况可以用十进制数字（或者十六进制数字）表示(见下表)。同时pair-end reads成对出现，如果一条read的flag是99，那边对应的另一个read就是147(见下图)。如果比对情况有多个，将多个比对情况所代表的十进制数字加和就是这一行的FLAG。</li>
</ul>
<table>
<thead>
<tr>
<th align="left">Value 10</th>
<th align="left">Value 16</th>
<th align="left">Description</th>
<th align="center">99</th>
<th align="center">83</th>
<th align="center">147</th>
<th align="center">163</th>
</tr>
</thead>
<tbody><tr>
<td align="left">1</td>
<td align="left">0x1</td>
<td align="left">read paired</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">2</td>
<td align="left">0x2</td>
<td align="left">read mapped in proper pair</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">4</td>
<td align="left">0x4</td>
<td align="left">read unmapped</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">8</td>
<td align="left">0x8</td>
<td align="left">mate unmapped</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">16</td>
<td align="left">0x10</td>
<td align="left">read reverse strand</td>
<td align="center"></td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center"></td>
</tr>
<tr>
<td align="left">32</td>
<td align="left">0x20</td>
<td align="left">mate reverse strand</td>
<td align="center">+</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">64</td>
<td align="left">0x40</td>
<td align="left">first in pair</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">128</td>
<td align="left">0x80</td>
<td align="left">second in pair</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">+</td>
<td align="center">+</td>
</tr>
<tr>
<td align="left">256</td>
<td align="left">0x100</td>
<td align="left">not primary alignment</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">512</td>
<td align="left">0x200</td>
<td align="left">read fails platform&#x2F;vendor quality checks</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">1024</td>
<td align="left">0x400</td>
<td align="left">read is PCR or optical duplicate</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr>
<td align="left">2048</td>
<td align="left">0x800</td>
<td align="left">supplementary alignment</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
</tbody></table>
<img src="/2024/01/25/4.bamFormat/sam_flag.png" srcset="/img/loading.gif" lazyload class="">
<ul>
<li><p>图中first read的FLAG是99（1+2+32+64），则表明该read</p>
<figure class="highlight applescript"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs applescript"><span class="hljs-number">1</span>：该<span class="hljs-built_in">read</span>是成对的paired reads中的一个<br><span class="hljs-number">2</span>：paired reads中每个都正确比对到参考序列上<br><span class="hljs-number">32</span>：与该<span class="hljs-built_in">read</span>成对的matepair <span class="hljs-built_in">read</span>其反向互补序列能够比对到参考序列<br><span class="hljs-number">64</span>：在paired reads中，该<span class="hljs-built_in">read</span>是与参考序列比对的第一条<br></code></pre></td></tr></table></figure></li>
<li><p>同时该reads对应的secnod read的FLAG必然是147（1+2+16+128），则表示</p>
<figure class="highlight applescript"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs applescript"><span class="hljs-number">1</span>：该<span class="hljs-built_in">read</span>是成对的paired reads中的一个<br><span class="hljs-number">2</span>：paired reads中每个都正确比对到参考序列上<br><span class="hljs-number">16</span>：该<span class="hljs-built_in">read</span>其反向互补序列能够比对到参考序列<br><span class="hljs-number">128</span>：在paired reads中，该<span class="hljs-built_in">read</span>是与参考序列比对的第二条。（也就是说，该<span class="hljs-built_in">read</span>是read2的反向互补序列）<br></code></pre></td></tr></table></figure>
<figure class="highlight sql"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs sql">只有一条<span class="hljs-keyword">reads</span>没有比对上：<span class="hljs-number">73</span>, <span class="hljs-number">133</span>, <span class="hljs-number">89</span>, <span class="hljs-number">121</span>, <span class="hljs-number">165</span>, <span class="hljs-number">181</span>, <span class="hljs-number">101</span>, <span class="hljs-number">117</span>, <span class="hljs-number">153</span>, <span class="hljs-number">185</span>, <span class="hljs-number">69</span>, <span class="hljs-number">137</span><br>两条<span class="hljs-keyword">reads</span>都没有比对上：<span class="hljs-number">77</span>、<span class="hljs-number">141</span><br>比对上了，方向也对，并且在插入片段大小范围内：<span class="hljs-number">99</span>, <span class="hljs-number">147</span>, <span class="hljs-number">83</span>, <span class="hljs-number">163</span><br>比对上了，也在插入片段大小范围内， 但是方向不对：<span class="hljs-number">67</span>, <span class="hljs-number">131</span>, <span class="hljs-number">115</span>, <span class="hljs-number">179</span><br>唯一配对，就是插入片段大小范围不对：<span class="hljs-number">81</span>, <span class="hljs-number">161</span>, <span class="hljs-number">97</span>, <span class="hljs-number">145</span>, <span class="hljs-number">65</span>, <span class="hljs-number">129</span>, <span class="hljs-number">113</span>, <span class="hljs-number">177</span><br></code></pre></td></tr></table></figure>
</li>
<li><p>值得注意的是，pair-end reads是指测序时生成的一对reads，分别来自同一个DNA片段的两端。假设read的名字是r001，如果read1和read2都能比对上参考序列，那么r001在SAM文件中会出现两次。如果r001的read1比对到参考序列的两个不同位置，那么r001的名字会出现三次。如果read1成功比对，而read2没有比对上，r001仍会出现两次，但其中一个r001的第三列会标记为”*”，表示未比对。因此，在pair-end测序中，read1和read2的文件同时mapping，相同reads的ID最少会出现2次。</p>
</li>
<li><p>以下网站可以通过输入FLAG值，直接找出该FLAG是那些FLAG的加和：Decoding SAM flags（<a target="_blank" rel="noopener" href="https://broadinstitute.github.io/picard/explain-flags.html%EF%BC%89">https://broadinstitute.github.io/picard/explain-flags.html）</a></p>
</li>
</ul>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs brainfuck"><span class="hljs-literal">-------------------------------------------------------------</span><br><span class="hljs-literal">---</span><span class="hljs-comment">R1</span><span class="hljs-literal">---</span>&gt; <span class="hljs-comment">99</span>                  <span class="hljs-literal">--------</span><span class="hljs-comment">R2</span><span class="hljs-literal">------</span>&gt;  <span class="hljs-comment">163</span><br><br>          &lt;<span class="hljs-literal">----</span><span class="hljs-comment">R2</span><span class="hljs-literal">----</span>  <span class="hljs-comment">147</span>            &lt;<span class="hljs-literal">--------</span><span class="hljs-comment">R1</span><span class="hljs-literal">------</span>  <span class="hljs-comment">83</span><br><span class="hljs-literal">-------------------------------------------------------------</span> <span class="hljs-comment">reverse strand</span><br></code></pre></td></tr></table></figure>

<h3 id="3-RNAME"><a href="#3-RNAME" class="headerlink" title="3.RNAME"></a>3.RNAME</h3><ul>
<li>比对上的参考序列的名字，该名字出现在Header section的@SQ行的SN标识中，如果该read没有比对上，也就是说该read在参考序列上没有坐标，那么这一列则用””表示，那么这一行的POS和CIGAR列也会是””。</li>
</ul>
<h3 id="4-POS"><a href="#4-POS" class="headerlink" title="4.POS"></a>4.POS</h3><ul>
<li>该read比对到参考序列RNAME的最左侧的位置坐标，最小为1（1-based leftmost）。也是CIGAR中第一个比对标识“M”对应的最左侧碱基在参考序列的位置，未比对上的read在参考序列中没有坐标，此列标识为“0”。</li>
</ul>
<h3 id="5-MAPQ"><a href="#5-MAPQ" class="headerlink" title="5.MAPQ"></a>5.MAPQ</h3><ul>
<li>对应参考序列的质量（MAPing Quality），比对的质量分数，越高说明该read比对到参考基因组上的位置越准确。其值等于-10log10Probility{错配概率}，得出值后四舍五入的整数就是MAPQ值。例如，MAPQ为20，即Q20，错误率为0.01，20&#x3D;-10log10(0.01) &#x3D; -10*(-2)。如果该值是255，则说明对应质量无效。</li>
</ul>
<h3 id="6-CIGAR"><a href="#6-CIGAR" class="headerlink" title="6.CIGAR"></a>6.CIGAR</h3><p>CIGAR标识符描述read与参考序列的比对具体情况信息。read中每个碱基的比对情况，主要有标识符: M表示匹配、I表示插入、D表示删除、N表示内含子和D类似、S表示替换、H表示剪切。</p>
<ul>
<li><p>M: alignment match (can be a sequence match or mismatch)<br>read上的碱基与参考序列“RNAME”完全匹配，碱基一一对应，包括了正确匹配与错误匹配</p>
</li>
<li><p>I: insertion to the reference<br>read上的碱基相对于参考序列“RNAME”有插入现象（如下）：</p>
<figure class="highlight objectivec"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs objectivec">REF:  <span class="hljs-built_in">CACGATCA</span>**GACCGATACGTCCGA <br>READ1:  <span class="hljs-built_in">CGATCAGAGACCGATA</span> <br><span class="hljs-built_in">CIGAR</span>：<span class="hljs-number">6</span>M2I8M<br></code></pre></td></tr></table></figure></li>
<li><p>D: deletion from the reference<br>read上的碱基相对于参考序列“RNAME”有删除现象（如下）：</p>
<figure class="highlight avrasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs avrasm"><span class="hljs-symbol">REF:</span> AGCTAGCATCGTGTCGCCCGTCTAGCATACGC<br><span class="hljs-symbol">READ:</span>             TCGCCCGT-TAGCAT<br>CIGAR：<span class="hljs-number">8</span>M1D6M<br></code></pre></td></tr></table></figure></li>
<li><p>N: skipped region from the reference<br>read上的碱基相对于参考序列“RNAME”存在连续没有比对上的空缺，这些空缺用N来表示，跟“D”相似但远比“D”缺失的更多，这种比对类型也叫“Spliced alignment”，常见cDNA与参考序列比对（如下）:”…”表示intron</p>
<figure class="highlight lasso"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs lasso">REF:AGCATCGTGTCGCCCGTCTAGCATACGCATGATCGACTGTCAGCTAGTCAGACTA<br>READ:     GTGTAACCC<span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span><span class="hljs-params">...</span>..TCAGAATA<br>CIGAR：<span class="hljs-number">9</span>M32N8M<br></code></pre></td></tr></table></figure></li>
<li><p>S: soft clipping (clipped sequences present in SEQ)</p>
</li>
<li><p>H: hard clipping (clipped sequences NOT present in SEQ)<br>read的开头或者结尾部分没有比对到参考序列”RNAME”上，但这部分未比对上的连续序列仍保留在sam文件的该read序列中，用“S”来表示；如果未保留，则用“H”表示，也即“hard cliping”（如下所示，也可同图2中r003的比对CIGAR中看出）</p>
<figure class="highlight gcode"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs gcode">REF: AGCTAGCATCGTGTCGCCCGTCTAGCATACGCAT<br>READ:          gggTCGCCCGT-TAGCATgggg<br>CIGAR：<span class="hljs-number">3</span>S<span class="hljs-number">8</span><span class="hljs-name">M1</span>D<span class="hljs-number">6</span><span class="hljs-name">M4</span>S （在sam中存储为GGGGTGTAACCGACTAGGGGG）<br>CIGAR：<span class="hljs-number">3</span>H<span class="hljs-number">8</span><span class="hljs-name">M1</span>D<span class="hljs-number">6</span><span class="hljs-name">M4</span>H （在sam中存储为GTGTAACCGACTAG）<br></code></pre></td></tr></table></figure></li>
<li><p>P: padding (silent deletion from padded reference)<br>多条read比对到参考序列的同一位置时，如果不同read单独同该参考序列比对时，参考序列的情况也不同，比如下方READ1同参考序列比对时，“GA”属于插入（6M2I8M），READ2同参考序列比对时，“A”属于插入（4M1I9M ），READ3同参考序列完全匹配（10M），没有插入，但是三条read之前却没有可比性。因此，当参考序列“比对情况包含完整”且序列唯一时，所有read同时进行比对，read3这种原本没有插入却默认插入的比对称之为Padded alignment，这种情况用“P”表示。</p>
<figure class="highlight nestedtext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><code class="hljs nestedtext"><span class="hljs-attribute">REF</span><span class="hljs-punctuation">:</span> <span class="hljs-string"> CACGATCA**GACCGATACGTCCGA</span><br><span class="hljs-attribute">READ1</span><span class="hljs-punctuation">:</span> <span class="hljs-string"> CGATCAGAGACCGATA </span><br><span class="hljs-attribute">READ2</span><span class="hljs-punctuation">:</span> <span class="hljs-string">   ATCA*AGACCGATAC </span><br><span class="hljs-attribute">READ3</span><span class="hljs-punctuation">:</span> <span class="hljs-string">  GATCA**GACCG  </span><br><span class="hljs-attribute">The padded CIGAR are different</span><span class="hljs-punctuation">: </span><br><span class="hljs-attribute">READ1</span><span class="hljs-punctuation">:</span> <span class="hljs-string">6M2I8M </span><br><span class="hljs-attribute">READ2</span><span class="hljs-punctuation">:</span> <span class="hljs-string">4M1P1I9M  </span><br><span class="hljs-attribute">READ3</span><span class="hljs-punctuation">:</span> <span class="hljs-string">5M2P5M</span><br></code></pre></td></tr></table></figure></li>
<li><p>&#x3D;：sequence match 正确匹配</p>
</li>
<li><p>X：sequence mismatch 错误匹配</p>
</li>
</ul>
<blockquote>
<p>perl CIGAR代码 为了检测每个read的甲基化情况</p>
</blockquote>
  <figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><code class="hljs perl"><span class="hljs-keyword">my</span> @f=<span class="hljs-keyword">split</span>(<span class="hljs-regexp">/\s+/</span>,$line);<br><br><span class="hljs-keyword">my</span> $flag = $f[<span class="hljs-number">1</span>];<br><span class="hljs-keyword">my</span> $chr = $f[<span class="hljs-number">2</span>];<br><span class="hljs-keyword">my</span> $pos = $f[<span class="hljs-number">3</span>];<br><span class="hljs-keyword">my</span> $cigar_2 = $f[<span class="hljs-number">5</span>];<br><br><span class="hljs-comment"># parsing CIGAR string</span><br><span class="hljs-keyword">my</span> @len = <span class="hljs-keyword">split</span> (<span class="hljs-regexp">/\D+/</span>,$cigar_2); <span class="hljs-comment"># storing the length per operation</span><br><span class="hljs-keyword">my</span> @ops = <span class="hljs-keyword">split</span> (<span class="hljs-regexp">/\d+/</span>,$cigar_2); <span class="hljs-comment"># storing the operation</span><br><span class="hljs-keyword">shift</span> @ops; <span class="hljs-comment"># remove the empty first element</span><br><span class="hljs-keyword">die</span> <span class="hljs-string">&quot;CIGAR string contained a non-matching number of lengths and operations ($cigar_2)\n&quot;</span> <span class="hljs-keyword">unless</span> (<span class="hljs-keyword">scalar</span> @len == <span class="hljs-keyword">scalar</span> @ops);<br><br><span class="hljs-comment">### determining end position of the read</span><br><span class="hljs-keyword">my</span> $readPos = <span class="hljs-number">0</span>;<br><span class="hljs-keyword">my</span> $genomePos = $pos;<br><br><span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $index(<span class="hljs-number">0</span>..$#len)&#123;<br><br>	<span class="hljs-keyword">my</span> $lengthSeg = $len[$index];<br><br>	<span class="hljs-keyword">if</span> ($ops[$index] eq <span class="hljs-string">&#x27;M&#x27;</span>)&#123;  <span class="hljs-comment"># standard matching bases</span><br><br>		$readPos += $lengthSeg;<br>		$genomePos += $lengthSeg;<br>	  <br>	&#125;<br>	<span class="hljs-keyword">elsif</span>($ops[$index] eq <span class="hljs-string">&#x27;I&#x27;</span>)&#123; <span class="hljs-comment"># insertions do not affect the end position</span><br>	  <br>		$readPos += $lengthSeg;<br><br>	&#125;<br>	<span class="hljs-keyword">elsif</span>($ops[$index] eq <span class="hljs-string">&#x27;D&#x27;</span>)&#123; <span class="hljs-comment"># deletions do affect the end position</span><br>	  $genomePos += $lengthSeg;<br><br>	&#125;<span class="hljs-keyword">elsif</span>($ops[$index] eq <span class="hljs-string">&#x27;S&#x27;</span>)&#123; <span class="hljs-comment"># deletions do affect the end position</span><br>	    $readPos += $lengthSeg;<br>	&#125;<br>&#125;<br></code></pre></td></tr></table></figure>


<h3 id="7-RNEXT"><a href="#7-RNEXT" class="headerlink" title="7.RNEXT"></a>7.RNEXT</h3><ul>
<li>该read的mate read比对上的参考序列的名字，该名字出现在Header section的@SQ行的SN标识中，</li>
<li>如果和该read所在行的第三列“RNAME”一样，则用“&#x3D;”表示，说明这对read比对到了同一条参考序列上；</li>
<li>如果mate read没有比对上，第七列则用“*”表示；</li>
<li>如果这对read没有比对到同一条参考序列，那么这一列则是mate read所在行第三列的“RNAME”。</li>
</ul>
<h3 id="8-PNEXT"><a href="#8-PNEXT" class="headerlink" title="8.PNEXT"></a>8.PNEXT</h3><ul>
<li>该read的mate read比对到的参考序列“RNAME”<strong>最左侧</strong>的位置坐标，也是mate read CIGAR中第一个比对标识“M”对应的最左侧碱基在参考序列的位置，未比对上的read在参考序列中没有坐标，此列标识为“0”。</li>
</ul>
<h3 id="9-ISIZE"><a href="#9-ISIZE" class="headerlink" title="9.ISIZE"></a>9.ISIZE</h3><ul>
<li><p>表示pair read完全匹配到同一条参考序列时，两个read之间的长度，可简单理解为测序文库的长度。这个定义有两种情况（虚线表示未比对上的序列，即soft-clipped bases）：</p>
</li>
<li><p>第一种情况如图左所示，Read1和Read2比对到同一条参考序列，此时ISIZE即为Read2的最右侧比对坐标减去Read1最左侧比对坐标；</p>
</li>
<li><p>第二种情况如图右所示，由于至今没有明确的定义和共识，因此ISIZE可以是TLEN#1，也可以是TLEN#2，视情况而定。</p>
</li>
</ul>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs brainfuck">                                            <span class="hljs-comment">|</span>&lt;<span class="hljs-literal">--</span><span class="hljs-comment">TLEN#1</span><span class="hljs-literal">----</span>&gt;<span class="hljs-comment">|</span><br><span class="hljs-literal">-------------------------------</span>   <span class="hljs-literal">-----------------------------------</span><br><span class="hljs-comment">|</span><span class="hljs-literal">---</span><span class="hljs-comment">R1</span><span class="hljs-literal">---</span>&gt; <span class="hljs-comment">99                               |</span><span class="hljs-literal">---</span><span class="hljs-comment">R1</span><span class="hljs-literal">-----------------</span>&gt; <span class="hljs-comment">99</span> <br><br>                &lt;<span class="hljs-literal">----</span><span class="hljs-comment">R2</span><span class="hljs-literal">----</span><span class="hljs-comment">| 147</span>     &lt;<span class="hljs-literal">----------</span><span class="hljs-comment">R2</span><span class="hljs-literal">---------</span><span class="hljs-comment">|  147</span><br><span class="hljs-literal">-------------------------------</span>   <span class="hljs-literal">------------------------------------</span><br><span class="hljs-comment">|</span>&lt;<span class="hljs-literal">-------</span><span class="hljs-comment">TLEN #1/#2</span><span class="hljs-literal">-------</span>&gt;<span class="hljs-comment">|         |</span>&lt;<span class="hljs-literal">-------</span><span class="hljs-comment">TLEN #2</span><span class="hljs-literal">--------------</span>&gt;<span class="hljs-comment">|</span><br></code></pre></td></tr></table></figure>

<h3 id="10-SEQ"><a href="#10-SEQ" class="headerlink" title="10.SEQ"></a>10.SEQ</h3><ul>
<li>存储的序列，没有存储，此列则用“*”标识。该序列的长度<strong>一定等于</strong>CIGAR标识中“M”，“I”，“S”，“&#x3D;”，“X”标识的碱基长度之和。</li>
</ul>
<h3 id="11-QUAL"><a href="#11-QUAL" class="headerlink" title="11.QUAL"></a>11.QUAL</h3><ul>
<li><p>Base quality score:序列的每个碱基对应一个碱基质量字符，每个碱基质量字符对应的ASCII码值减去33（Sanger Phred-33 质量值体系），即为该碱基的测序质量得分（Phred Quality Score）。不同Phred Quality Score代表不同的碱基测序错误率，如Phred Quality Score值为20和30分别表示碱基测序错误率为1%和0.1%。（ASCII码从0-127，为什么很多Phred评分从33开始呢？因为33是第一个肉眼可见的字符!，在此之前的都是控制符，比如空格、制表等，不方便用这些控制符来表示一个碱基的质量。）</p>
</li>
<li><p>假设一个碱基测序错误的概率是P，那么此时对应的Phred quality score（常用Q表示）</p>
<blockquote>
<p>$Q &#x3D; -10log_{10}(P)$</p>
</blockquote>
</li>
<li><p>假设100个碱基中有1个出现错误，即P&#x3D;0.01，根据上式可以计算，此时Q&#x3D;20；同理，当测序错误率为0.001时，此时Q&#x3D;30。<br>测序结果中的每一个碱基都有这么一个Phred评分。为了便于描述，人们习惯上将Phred评分对应一个ASCII码，这样每一个碱基的质量都可以用一个字符来表述了。</p>
</li>
</ul>
<table>
<thead>
<tr>
<th align="left">P</th>
<th align="center">Q (Phred quality score)</th>
<th align="center">Sanger Phred</th>
<th align="left">ASCII</th>
</tr>
</thead>
<tbody><tr>
<td align="left">0.01</td>
<td align="center">20</td>
<td align="center">53</td>
<td align="left"><strong>5</strong></td>
</tr>
<tr>
<td align="left">0.001</td>
<td align="center">30</td>
<td align="center">63</td>
<td align="left"><strong>?</strong></td>
</tr>
<tr>
<td align="left">0.0001</td>
<td align="center">40</td>
<td align="center">73</td>
<td align="left"><strong>I</strong></td>
</tr>
<tr>
<td align="left">0.00001</td>
<td align="center">50</td>
<td align="center">83</td>
<td align="left"><strong>S</strong></td>
</tr>
</tbody></table>
<h3 id="12-可选的区域"><a href="#12-可选的区域" class="headerlink" title="12.可选的区域"></a>12.可选的区域</h3><ul>
<li>格式类似AS:i:-1 XN:i:0 XM:i:1 XO:i:0 XG:i:0 NM:i:1 MD:Z:35T0 YT:Z:UU<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><code class="hljs stylus">AS:<span class="hljs-selector-tag">i</span>  匹配的得分<br>XS:<span class="hljs-selector-tag">i</span>  第二好的匹配的得分<br>YS:<span class="hljs-selector-tag">i</span>  mate 序列匹配的得分<br>XN:<span class="hljs-selector-tag">i</span>  在参考序列上模糊碱基的个数<br>XM:<span class="hljs-selector-tag">i</span>  错配的个数<br>XO:<span class="hljs-selector-tag">i</span>  <span class="hljs-attribute">gap</span> open的个数<br>XG:<span class="hljs-selector-tag">i</span>  <span class="hljs-attribute">gap</span> 延伸的个数<br>NM:<span class="hljs-selector-tag">i</span>  经过编辑的序列<br>YF:<span class="hljs-selector-tag">i</span>  说明为什么这个序列被过滤的字符串<br>YT:Z 值为UU表示不是pair中一部分(单末端？)、<span class="hljs-built_in">CP</span>(是pair且可以完美匹配)、<span class="hljs-built_in">DP</span>(是pair但不能很好的匹配)、<span class="hljs-built_in">UP</span>(是pair但是无法比对到参考序列上)<br>MD:Z  代表序列和参考序列错配的字符串<br></code></pre></td></tr></table></figure></li>
</ul>
<h2 id="3-samtools-软件使用"><a href="#3-samtools-软件使用" class="headerlink" title="3. samtools 软件使用"></a>3. samtools 软件使用</h2><h3 id="常用比对软件"><a href="#常用比对软件" class="headerlink" title="常用比对软件"></a>常用比对软件</h3><figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-comment">#使用BWA进行比对</span><br><span class="hljs-attribute">bwa</span> mem -M ref.fa r1.fq.gz r2.fq.gz -t <span class="hljs-number">4</span> | samtools view -Sb - | samtools sort -o sorted_reads.bam<br><br><span class="hljs-comment">#使用Bowtie2进行比对</span><br><span class="hljs-attribute">bowtie2</span> -x genome -<span class="hljs-number">1</span> read1.fq.gz -<span class="hljs-number">2</span> read2.fq.gz -p <span class="hljs-number">4</span> | samtools view -Sb - | samtools sort -o sorted_reads.bam<br></code></pre></td></tr></table></figure>

<h3 id="sort：sort-alignment-file"><a href="#sort：sort-alignment-file" class="headerlink" title="sort：sort alignment file"></a>sort：sort alignment file</h3><ul>
<li>本命令对bam文件中的序列进行排序，默认下是按序列在fasta文件中的顺序（即header）和序列从左往右的位点排序。</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools <span class="hljs-built_in">sort</span> [-T out.prefix][-@ threads][-m maxMem][-o out.sorted.bam][in.bam]<br>samtools <span class="hljs-built_in">sort</span> -@ 10 -o abc.sorted.bam abc.bam<br>samtools <span class="hljs-built_in">sort</span> -m 500000000 -o abc.sorted.bam abc.bam <span class="hljs-comment"># 500M dang abc.bam文件比较大，split的文件比较多，比如200多个。</span><br></code></pre></td></tr></table></figure>
<h3 id="merge：-merge-sorted-alignments"><a href="#merge：-merge-sorted-alignments" class="headerlink" title="merge： merge sorted alignments"></a>merge： merge sorted alignments</h3><ul>
<li><p>本命令将多个排好序的bam比对文件进行合并，产生一个排好序的bam输出文件(合并后的文件不需要再进行sort)，这个文件含有所有的输入记录，并且保留了他们原来的顺序。</p>
<blockquote>
<p>samtools merge [out.bam][in1.bam][in2.bam][in3.bam]</p>
</blockquote>
</li>
</ul>
<h3 id="index：index-alignment"><a href="#index：index-alignment" class="headerlink" title="index：index alignment"></a>index：index alignment</h3><ul>
<li><p>本命令对bam文件建立索引并产生后缀为.bai的文件，用于快速的随机处理。很多后续分析的过程需要有bai文件的存在，特别是显示序列比对情况下，比如samtool的tview命令等。#必须对bam文件进行默认情况下的排序后，才能进行index。否则会报错</p>
<blockquote>
<p>samtools index aln.sorted.bam</p>
</blockquote>
</li>
</ul>
<h3 id="view：SAMBAMCRAM-conversion"><a href="#view：SAMBAMCRAM-conversion" class="headerlink" title="view：SAM&lt;-&gt;BAM&lt;-&gt;CRAM conversion"></a>view：SAM&lt;-&gt;BAM&lt;-&gt;CRAM conversion</h3><ul>
<li>本命令将sam文件转换成bam文件；然后对bam文件进行各种操作，比如数据的排序和提取(这些操作是对bam文件进行的，不能对sam文件进行该操作)；最后将排序或提取得到的数据输出为bam或sam格式。bam文件优点：bam文件为二进制文件，占用的磁盘空间比sam文本文件小；利用bam二进制文件的运算速度快。</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools view -b abc.sam &gt; abc.bam<br>samtools view -b abc.sam -o abc.bam<br>samtools view abc.bam chr1 |less<br>samtools view abc.bam chr1:1000000-2000000 |less<br></code></pre></td></tr></table></figure>

<h3 id="tview"><a href="#tview" class="headerlink" title="tview"></a>tview</h3><ul>
<li>直观显示reads比对到基因组的情况，与IGV类似需要先sort和index</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools tview abc.sorted.bam hg19.fasta<br></code></pre></td></tr></table></figure>
<ul>
<li>press <strong>g</strong> to check the alignment start from a region in the format like chr10:10,000,000 or &#x3D;10,000,000 when viewing the same reference sequence.</li>
<li>c, C, n …</li>
</ul>
<h3 id="depth"><a href="#depth" class="headerlink" title="depth"></a>depth</h3><ul>
<li>统计每个碱基位点的测序深度;需要使用重定向定义输出文件；要使用构建过索引的bam Usage: bam2depth [-r reg] [-q baseQthres] [-Q mapQthres] [-b in.bed] […] -r：（region）加染色体号； -q：要求测序碱基质量最低值； -Q：要求比对的质量最低值<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools depth abc.sorted.bam &gt;abc.depth<br></code></pre></td></tr></table></figure></li>
</ul>
<h2 id="参考文献"><a href="#参考文献" class="headerlink" title="参考文献"></a>参考文献</h2><ul>
<li>Li H, Handsaker B, Wysoker A, et al. The Sequence Alignment&#x2F;Map format and SAMtools[J]. Bioinformatics, 2009, 25(16): 2078-2079.</li>
<li><a target="_blank" rel="noopener" href="https://samtools.github.io/hts-specs/SAMv1.pdf">https://samtools.github.io/hts-specs/SAMv1.pdf</a></li>
<li><a target="_blank" rel="noopener" href="https://www.samformat.info/sam-format-flag">https://www.samformat.info/sam-format-flag</a></li>
<li><a target="_blank" rel="noopener" href="https://broadinstitute.github.io/picard/explain-flags.html">https://broadinstitute.github.io/picard/explain-flags.html</a></li>
</ul>

                
              </div>
            
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/" class="category-chain-item">生物信息学</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>4.Sam/Bam文件格式详解</div>
      <div>https://xiaoqiangq.github.io/2024/01/25/4.bamFormat/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>作者</div>
          <div>Qiang</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>发布于</div>
          <div>2024年1月25日</div>
        </div>
      
      
      
        <div class="license-meta-item">
          <div>许可协议</div>
          <div>
            
              
              
                <a class="print-no-link" target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
                  <span class="hint--top hint--rounded" aria-label="BY - 署名">
                    <i class="iconfont icon-by"></i>
                  </span>
                </a>
              
            
          </div>
        </div>
      
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2024/01/26/5.VCFFormat/" title="5.VCF文件格式详解">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">5.VCF文件格式详解</span>
                        <span class="visible-mobile">上一篇</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2024/01/24/3.fastqFormat/" title="3.Fastq/Fasta文件格式详解">
                        <span class="hidden-mobile">3.Fastq/Fasta文件格式详解</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>
  </div>
</div>





  



  



  



  



  







    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <a href="https://www.cpu.edu.cn/" target="_blank" rel="nofollow noopener"><span>CPU</span></a> <i class="iconfont icon-love"></i> <a href="https://ygl.cpu.edu.cn/gmis5/dsfc/dsfcgrxx/545CA439547A89F80D33CDEEC7CB033C" target="_blank" rel="nofollow noopener"><span>Qiang</span></a> 
    </div>
  
  
  
  
</div>

  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.4/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>




  
    <script  src="/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.20.1/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init(Object.assign({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      scrollSmooth    : true,
      includeTitleTags: true,
      headingsOffset  : -boardTop,
    }, CONFIG.toc));
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }

    Fluid.events.registerRefreshCallback(function() {
      if ('tocbot' in window) {
        tocbot.refresh();
        var toc = jQuery('#toc');
        if (toc.length === 0 || !tocbot) {
          return;
        }
        if (toc.find('.toc-list-item').length > 0) {
          toc.css('visibility', 'visible');
        }
      }
    });
  });
</script>


  <script src=https://lib.baomitu.com/clipboard.js/2.0.11/clipboard.min.js></script>

  <script>Fluid.plugins.codeWidget();</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));

    Fluid.events.registerRefreshCallback(function() {
      if ('anchors' in window) {
        anchors.removeAll();
        var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
        var res = [];
        for (var item of el) {
          res.push('.markdown-body > ' + item.trim());
        }
        if (CONFIG.anchorjs.placement === 'left') {
          anchors.options.class = 'anchorjs-link-left';
        }
        anchors.add(res.join(', '));
      }
    });
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/js/local-search.js" ></script>

  <script defer src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">博客在允许 JavaScript 运行的环境下浏览效果更佳</div>
  </noscript>
</body>
</html>
